#!/bin/bash
# The interpreter used to execute the script

#“#SBATCH” directives that convey submission options:

#SBATCH --job-name=array_trim
#SBATCH --mail-type=END
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=5g
#SBATCH --time=2:00:00
#SBATCH --account=bfoxman1
#SBATCH --partition=standard
#SBATCH --output=/scratch/bfoxman_root/bfoxman/blostein/meta_CAVITIES/Code/CleanPipe/%x-%j.log

echo "All jobs in this array have:"
echo "- SLURM_ARRAY_JOB_ID=${SLURM_ARRAY_JOB_ID}"
echo "- SLURM_ARRAY_TASK_COUNT=${SLURM_ARRAY_TASK_COUNT}"
echo "- SLURM_ARRAY_TASK_MIN=${SLURM_ARRAY_TASK_MIN}"
echo "- SLURM_ARRAY_TASK_MAX=${SLURM_ARRAY_TASK_MAX}"
 
echo "This job in the array has:"
echo "- SLURM_JOB_ID=${SLURM_JOB_ID}"
echo "- SLURM_ARRAY_TASK_ID=${SLURM_ARRAY_TASK_ID}"

#module load
module load Bioinformatics
module load trimmomatic
module load fastqc/0.11.8
module load bowtie2
module load samtools
module load bedtools2

#set nextera file
nextera=/scratch/bfoxman_root/bfoxman/blostein/meta_CAVITIES/Files/adapters/NexteraPE-PE.fa

#set input and output directory
in=/scratch/bfoxman_root/bfoxman/blostein/meta_CAVITIES/Files/OGFiles/ 
out=/scratch/bfoxman_root/bfoxman/blostein/meta_CAVITIES/Files/CleanPipe/PostTrim

#Files for input
FILES1=($(ls -1 ${in}*R1*.fastq.gz))
FILES2=($(ls -1 ${in}*R2*.fastq.gz))
FILENAME1=${FILES1[$SLURM_ARRAY_TASK_ID]}
FILENAME2=${FILES2[$SLURM_ARRAY_TASK_ID]}

/scratch/bfoxman_root/bfoxman/blostein/meta_CAVITIES/Code/CleanPipe/metagenomeTrimmomatic.sh ${FILENAME1} ${FILENAME2} 0 ${nextera} ${out}
