---
title: 'Testing the ecological hypothesis: Longitudinal case-control study of early
  childhood caries and salivary microbiome assembly'
author: "Freida Blostein"
date: "`r Sys.Date()`"
output:
  word_document: default
  html_document: default
---

```{r, echo = FALSE, include=FALSE, warning=FALSE, message=FALSE}
#prj directory
prjdir<-gsub('/Code', '', here::here())
data.out<-file.path(prjdir, 'Data')
code.dir<-file.path(prjdir, 'Code')
output.out<-file.path(prjdir, 'Output')

#results
main.out<-file.path(output.out, 'Main')
supp.out<-file.path(output.out, 'Supplement')
dir.create(main.out)
dir.create(supp.out)

#wgs
wgs.path=file.path(prjdir, 'OGData', 'COHRA', 'WGS')

#options
keepDouble=TRUE

#a billion libraries
#data manipulation
library(dplyr)
library(tidyr)
library(purrr)
library(broom)
library(stringr)
library(stringi)
library(forcats)
library(readr)
library(tibble)
library(openxlsx)
library(compareGroups)
library(psych)

#data visualization
library(ggplot2)
library(ggalluvial)
library(ggpubr)
library(ggtree)
library(ggjoy)
library(ggrepel)
library(tidygraph); 
library(ggraph);
library(igraph)
library(ggvenn)
library(cowplot)
library(patchwork)
library(grid)
library(gridExtra)
library(gtable)

#microbiome
library(phyloseq)
library(phangorn)
library(ape)
library(msa)
library(SQMtools)

#knitr options
knitr::opts_chunk$set(echo = FALSE, include=FALSE, warning=FALSE, message=FALSE)

```

```{r read in data}
#read in phyloseq object
load(file.path(data.out, 'phyloseq', 'phyasv_meta.Rdata'))
ps_long=microbiome::transform(ps_meta_nodups, 'compositional')%>%psmelt()%>%dplyr::mutate(ScientificName=paste(Genus, ifelse(is.na(Species), OTU, Species), sep=' '))

#extract taxa info
taxa=as.data.frame(ps_meta_nodups@tax_table)%>%dplyr::mutate(ASV=row.names(.), ScientificName=paste(Genus, ifelse(is.na(Species), ASV, Species), sep=' '))

#metavisit data
load(file.path(data.out, 'Metadata', 'MetaVisit.Rdata'))
if(keepDouble==T){MV=MetaVisit}
if(keepDouble==F){MV=MetaVisit%>%filter(count!='Double; drop')}

#correct incident visits 
MV = MV %>% mutate(IncidentVisit2=IncidentVisit2_new)

#uscur info 
load(file.path(data.out, 'Metadata', 'myuscur.Rdata'))

#merge metavisit and uscur 
mv_usc = MV %>% 
  left_join(my_uscur)

#
my_uscur= my_uscur %>% 
  left_join(MV)
```

# Figure 1
```{r}
figure1=MetaVisit %>%
  filter(HasCleanSample==1 & Visit<=IncidentVisit2)%>%
  select(Visit, AgeAtExamMonths, UniqueID, Case, IncidentVisit2, sample_in_wgs)%>%
  mutate(sample_type=factor(
    case_when(Visit==IncidentVisit2 &
                sample_in_wgs=="Sample in shotgun sequenced subset"~'Incident visit sample (16S amplicon & WGS)', 
              Visit==IncidentVisit2 &
                sample_in_wgs!="Sample in shotgun sequenced subset"~'Incident visit sample (16S amplicon)',
                Visit<IncidentVisit2~'Pre-incident visit sample (16S amplicon)'), 
                levels=c('Pre-incident visit sample (16S amplicon)', 'Incident visit sample (16S amplicon)', 'Incident visit sample (16S amplicon & WGS)')))%>%
  unique()%>%
  ggplot(aes(x=AgeAtExamMonths))+
  geom_histogram(aes(fill=as.factor(sample_type)))+ theme_bw()+
  scale_x_continuous('Age (months)', breaks = c(0, 2, 9, 12, 24, 36, 48, 60))+
  scale_y_continuous('Count of saliva samples by \n incidence status and sequencing method')+
  theme(legend.position=c(0.8, 0.85))+
  scale_fill_manual('', values=c('grey', 'grey50', 'black'))+
  theme(text=element_text(size=20))

ggsave(filename=file.path(main.out, 'Figure1.pdf'), figure1, units=c('in'), width=16, height=6)
```


```{r load data supervised and unsupervised}
unsupervised_files<-c(list.files(file.path(data.out, 'Network'), full.names = T), list.files(file.path(data.out, 'CST'), full.names = T)[-1])

unsupervised_files%>%purrr::map(function(file){load(file, envir = .GlobalEnv)})

clr_network$module.data<-left_join(clr_network$module.data, MV)
hellinger_network$module.data<-left_join(hellinger_network$module.data, MV)

#pick most important cst 
main_cst=CST_k6
#merge with metavisit and rename NA as unclassified
CST1<-left_join(main_cst, MV %>% select(FoxCavID, UniquePersonID, UniqueID, BabySubjectID, Site, Case, IncidentVisit2, Visit, count))
levels(CST1$CST.8)<-c(levels(CST1$CST.8), 'Unclassified')
CST1[is.na(CST1$CST.8),'CST.8']<-"Unclassified"

#read in RF data
load(file.path(data.out, 'RF', 'rfResults_hellinger_iVisit_FALSE_duplicates_FALSE.Rdata'))

#wgs
load(file.path(data.out, 'MS', 'kegg_and_taxa_slim.Rdata'))
#load humann3 object
load(file.path(data.out, 'H3', 'Humann3.Rdata'))
```

# At 12- and 24-months of age, supervised random forest using the salivary bacteriome is more accurate at predicting ECC case status than *S. mutans* alone 

```{r randomForest - fig1}
#Fig1A: area under the curve roc
rf_roc<-rfAllEvals_hellinger$roc$data%>%
  #filter out S mutans models -- per reviewer
  filter(!str_detect(Group, 'S. mutans'))
fig1a<-ggplot(rf_roc, aes(x=FPR, y=SENS, color=Group, linetype=Group))+geom_line(size=1.2)+scale_color_manual(name='', values=c('black', 'grey63', 'black', 'grey63'))+scale_linetype_manual(name='', values=c(1, 1, 3, 3))+xlab('False positive rate')+ylab('True positive rate')+theme_classic()+theme(legend.text=element_text(face='italic'))+theme(legend.text=element_text(face='italic'))+guides(color=guide_legend(nrow=2,byrow=TRUE), linetype=guide_legend(nrow=2,byrow=TRUE))+theme(legend.position='bottom')+geom_abline(slope=1, intercept = 0, color='darkgrey')+coord_fixed()
#fig1b: importance of ASVs
#importance of ASVs
imp<-lapply(rfFits_hellinger[c(1, 3)], function(x) as.data.frame(x$finalModel$importance)%>%dplyr::mutate(ASV=row.names(x$finalModel$importance)))%>%bind_rows(.id="column_label")%>%dplyr::mutate(Visit=as.integer(gsub('V', '', gsub('All taxa', '', column_label))), Month=ifelse(Visit=='5', '12 month', '24 month'))
#info on >mean in case | control
greater_mean=ps_long %>% filter(Visit %in% c(5, 7))%>%dplyr::group_by(OTU, Visit, Case, ScientificName)%>%dplyr::summarise(mean=mean(Abundance))%>%pivot_wider(id_cols=c(OTU, Visit, ScientificName), names_from=Case, names_prefix='Abundance_', values_from=mean)%>%dplyr::mutate(GreaterMean=ifelse(Abundance_Case>Abundance_Control, 'Case', 'Control'), ASV=OTU)
#join 
imp=left_join(imp, greater_mean)
imp_top=rbind(imp%>%dplyr::group_by(Visit)%>%slice_max(n=10, order_by=MeanDecreaseGini), imp%>%filter(str_detect(ScientificName, 'mutans')))

imp_top=imp_top %>%
    group_by(ScientificName)%>%
    mutate(neg=ifelse(GreaterMean=='Case', -1, 1),
           max=max(MeanDecreaseGini)*neg)%>%
  mutate(SciName=fct_reorder(ScientificName, max))

sciname_order=imp_top%>%arrange(max)%>%pull(SciName)%>%unique()
#fig 1b plot
fig1b=imp_top%>%   
  ggplot(aes(x=fct_reorder(SciName, max), ymax=MeanDecreaseGini, ymin=0, color=GreaterMean)) +
    geom_point(aes(y=MeanDecreaseGini, color=GreaterMean, shape=GreaterMean), size=2)+
    geom_linerange()+coord_flip()+facet_grid(.~Month,scales='free_y')+
    theme_classic()+theme(axis.text.y=element_text(face='italic'))+
    theme(legend.text=element_text(face='italic'))+theme(legend.position = 'bottom')+
    scale_color_manual(name='Mean relative abundance greater in:\n', values=c('black', 'darkgrey'))+
    scale_y_continuous(name='Mean decrease Gini coefficient')+scale_x_discrete(name='')+
    scale_shape_manual(name='Mean relative abundance greater in:\n', values=c('triangle', 'square'))

#fig1c abundance of top important taxa
fig1c=ps_long%>%
  filter(Visit %in% c(5, 7) & OTU %in% imp_top$OTU)%>%
  dplyr::mutate(Month=ifelse(Visit==5, '12 month', '24 month'),
                SciName=factor(ScientificName, levels=rev(sciname_order)))%>%
  ggplot(aes(x=sqrt(Abundance), y=Case, fill=Case))+geom_joy()+facet_grid(SciName~Month, scales='free')+theme_classic()+theme(axis.text.y=element_blank(), axis.ticks.y = element_blank(), axis.title.y=element_blank(), strip.text.y=element_blank())+scale_fill_manual(name='ECC case status: ', values=c('black', 'darkgrey'))+theme(legend.position='bottom')+xlab('Square root(Relative abundance)')
#combine and save
fig1=fig1a+fig1b+fig1c+plot_annotation(tag_levels='A')
ggsave(filename=file.path(main.out, 'Figure2.eps'), fig1, units=c('in'), width=16, height=6)

rm(fig1, fig1a, fig1b, fig1c)

sm_12=paste(rfEvals_hellinger$`V5S. mutans only`$stdres$`Group 1`['AUC-ROC',], collapse=', (')
sm_24=paste(rfEvals_hellinger$`V7S. mutans only`$stdres$`Group 1`['AUC-ROC',], collapse=', (')
all_12=paste(rfEvals_hellinger$`V5All taxa`$stdres$`Group 1`['AUC-ROC',], collapse=', (')
all_24=paste(rfEvals_hellinger$`V7All taxa`$stdres$`Group 1`['AUC-ROC',], collapse=', (')
```

To test the ecological hypothesis, we compared the performance of a supervised random forest classifier when using either information on all taxa in the analytic subset or only the amplicon classified as S. mutans. We validate the identity of the amplicon identified as *S. mutans* using BLAST and data from the whole genome shotgun sequencing cohort (Supplementary text; Supplementary Tables XX; Supplementary Figure XX). The salivary bacteriome at both the 12- and 24-month visit(Figure 1A, AUC (95% CI): `r all_12`) and `r all_24`), respectively), accurately predicted future ECC case status. Classification using only the amplicon identified as *S. mutans* performed worse at both visits, and significantly worse at the 12-month visit (Figure 1A, AUC (95% CI): `r sm_12`) and `r sm_24`), respectively). Only 7 non-incident samples had detectable *S. mutans* at 12-months, although prevalence increased by 24-months and was significantly higher in future cases at the 24-month visit (Table 1). *S. mutans* prevalence and abundance was also elevated in incident samples at the 24- and 36-month visits (Supplemental Table XX)

The top 10 most important features in the all-taxa random-forest included members of common oral genera, including *Neisseria*, *Streptococcus*, *Rothia* and *Prevotella*, and were relatively consistent between the two time points (Figure 1B). Important features included taxa that were elevated in cases and taxa elevated in controls (Figure 1C). We validated that the two *Streptococcus* ASVs identified as important features in the random forest were not *S. mutans*; *Streptococcus* ASV8 is likely *S. salivarius*, and while Streptococcus ASV14 is closely related to *Streptococcus lactarius/peroris* (Supplementary text; Supplementary Tables XX & XX; Supplementary Figure XX). 

# Unsupervised clustering techniques identify similar guilds of co-occurring taxa, which associate with ECC 

Using Dirichlet multinomial community state typing, we identified 6 community state types. The proportion of samples assigned to each community state type over time and by case status is shown in Figure 2A; community state types were named according to the ASVs defining their separation. In both ECC cases and controls, community state types in the population transition from *Streptococcus*-dominated to more diverse community state types over time (Supplemental Figure XX). However, the progression of specific community state types was different in cases and controls.* Neisseria* ASV9 and *Hemophilus parainfluenzae* characterized community state types prevalent in cases, while *Streptococcus* ASV8 and *Veillonella* ASV5 characterized those prevalent in controls.  (Figure 2A, Table 1, Supplemental Table 1).

Using weighted co-occurrence network analysis, we identified 5 network modules of co-occurring ASVs. Network module relative abundances changed as children aged (Figure 2B Supplemental Figure XX). Network modules and community state types identified similar groupings of taxa: of the top ten taxa contributing to the separation of the community state types, nine were also among the top two most abundant taxa from the five network modules (Supplemental Figure XX).  As in the community state type analysis, a *Neisseria* ASV9 and *Haemophilus parainfluenzae* community was more abundant in controls; a *Veillonella* ASV5 and *Streptococcus* ASV8 community was more abundant in cases (Figure 2B; Table 1). The networks also provide additional structural information, including identifying low-abundance, highly central taxa, including *Fusobacterium periodonticum* (Figure 2C).  
The ECC-associated communities identified through unsupervised clustering were robust to varying important hyperparameters, namely choice of k community state types (k=4 vs 5 vs 6) and choice of normalization transform for weighted co-occurrence networks (Hellinger vs central-log) (Supplemental text, Supplemental Figures XX and XX). 


```{r visit labels}
#visit labels for plotting
visit_labels2<-c('2mos', '12mos', '24mos', '36mos', '48mos', '60mos')
names(visit_labels2)<-c('3', '5', '7', '8', '9', '10')
visit_labelsV2=visit_labels2
names(visit_labelsV2)<-paste0('V', c('3', '5', '7', '8', '9', '10'))
```

```{r name CSTs}
cst_recodes<-data.frame('CST.8'=as.factor(c(1, 2, 3, 4, 5, 6)), CSTlabels=factor(c('Neisseria ASV12 - Veillonella ASV5', 'H. parainfluenzae - Neisseria ASV9','Streptococcus ASV1 dominated w/ Gemella ASV2', 'Streptoccous ASV1 dominated w/ G. elegans', 'Gemella ASV2 - H. parainfluenzae - Neisseria ASV9', 'Streptococcus ASV8 - Neisseria ASV12'), levels=c('Streptococcus ASV1 dominated w/ Gemella ASV2', 'Streptoccous ASV1 dominated w/ G. elegans','Gemella ASV2 - H. parainfluenzae - Neisseria ASV9', 'Streptococcus ASV8 - Neisseria ASV12','H. parainfluenzae - Neisseria ASV9', 'Neisseria ASV12 - Veillonella ASV5')))
```

```{r unsupervised guilds cst - fig 2a}
CST<-CST1 %>% select(UniquePersonID, BabySubjectID, Site, Case, IncidentVisit2, Visit, CST.8) %>% filter(Visit%in%c(3, 5, 7, 8, 9 , 10))
CST.wide<-reshape2::dcast(CST, UniquePersonID+BabySubjectID+Site+Case+IncidentVisit2~Visit, value.var="CST.8")
data.long<-gather(CST.wide, "Visit","CST", 6:11)
data.long$Visit<-as.numeric(data.long$Visit)
#remove ended as case ended as control from fig 2a and replace with 'no microbiome data 
#data.long$CST2<-with(data.long, ifelse(IncidentVisit2<Visit, ifelse(Case=="Control", "Ended as control", "Ended as case"), CST))
data.long$CST2<-with(data.long, ifelse(IncidentVisit2<Visit, "No microbiome data", CST))
data.wide<- reshape2::dcast(data.long, UniquePersonID+BabySubjectID+Site+Case+IncidentVisit2 ~ Visit, value.var="CST2")
colnames(data.wide)[6:11]<-c( "V3", "V5", "V7","V8", "V9", "V10")
CST.wide<-data.wide %>% dplyr::count(V3, V5, V7, V8, V9, V10, Case)
CST.wide[is.na(CST.wide)]<- "No microbiome data"
CST.visit<-to_lodes_form(CST.wide, axes=1:6, id="cohort")
CST.visit2<-CST.visit %>% dplyr::group_by(Case, x) %>% dplyr::mutate(denom = sum(n))
CST.visit2<-CST.visit2 %>% dplyr::group_by(Case, x, stratum) %>% dplyr::mutate(num = sum(n))
CST.visit2<-CST.visit2 %>% ungroup() %>% dplyr::mutate(prop=num/denom)
CST.visit2<-CST.visit %>% group_by(Case, x) %>% dplyr::mutate(prop=n/sum(n))
CST.visit2$stratum<-recode_factor(CST.visit2$stratum, `3`='Streptococcus ASV1 dominated w/ Gemella ASV2',`5`='Gemella ASV2 - H. parainfluenzae-Neisseria ASV9', `2`='H. parainfluenzae - Neisseria ASV9', `4`='Streptoccous ASV1 dominated w/ G. elegans',  `6`='Streptococcus ASV8 - Neisseria ASV12',  `1`='Neisseria ASV12 - Veillonella ASV5', 'Ended as control'='Ended as control', 'Ended as case'='Ended as case', "No microbiome data"="No microbiome data", "Unclassified"="Unclassified at P>80%")

#count n samples for labels
count_n=CST.visit2%>%filter(!(stratum %in% c('Ended as control', 'Ended as case', "No microbiome data")))%>%select(-cohort, -prop, -stratum)%>%group_by(Case, x)%>%dplyr::summarise(n=sum(n))
count_n$stratum='Missed visit'; count_n$cohort=1

colors_with_case=c("thistle","deepskyblue", "royalblue2", "lightcoral",  "darkorange",   "darkorange3",  "royalblue4", "darkorange4", 'white', 'black')
colors_no_case=c("thistle","deepskyblue", "royalblue2", "lightcoral",  "darkorange",   "darkorange3",  'white', 'black',  "royalblue4", "darkorange4")

fig2a<-ggplot(CST.visit2,
       aes(x = x, stratum = stratum, alluvium = cohort,
           y = prop,
           fill = stratum)) +
    geom_text(data=count_n, aes(label=paste0('N=', n)), y=1, vjust=-0.5)+
    stat_stratum(width=.5)+geom_flow(alpha=.7)+geom_stratum(width=.5)+
    facet_grid(Case~.)+
    scale_x_discrete(name="Age (months)", labels=visit_labelsV2, expand=c(.1, .1))+
    scale_fill_manual(name="Community state type", breaks=levels(CST.visit2$stratum), values=colors_no_case, guide=guide_legend(nrow=2))+
    scale_y_continuous(name="Proportion", limits=c(0, 1.1), breaks=seq(0, 1, 0.2))+theme_classic()+
    theme(legend.position="bottom")+guides(fill=guide_legend(ncol =1,byrow=TRUE))+
    theme(text=element_text(size=20))+theme(axis.text.x=element_text(angle = 45, vjust = 0.5, hjust=.5))+
    theme(legend.text = element_text(face='italic', size=12))

cairo_ps(file.path(main.out, 'Figure3.eps'), family='Arial', width=10, height=8)
fig2a
dev.off()
cairo_pdf(file.path(main.out, 'Figure3.pdf'), family='Arial', width=10, height=8)
fig2a
dev.off()
```

```{r unsupervised guilds networks - fig 2b}
#Pick top genera to color
topGenus<-c('Streptococcus', 'Neisseria', 'Veillonella', 'Fusobacterium', 'Haemophilus', 'Rothia', 'Prevotella', 'Actinomyces' )
genus_colors=c(scales::hue_pal()(8), 'black')
genus_colors[3]="#910C00"
names(genus_colors)=c(topGenus, 'Other')
#Pick top taxa to label 
br_yel_cohra<-hellinger_network$intraStats%>%filter(module%in% c('brown', 'yellow'))%>%pull(ScientificName)
by_cohra=br_yel_cohra[!str_detect(br_yel_cohra, 'ASV')]
topTax<- c('Fusobacterium periodonticum', 'Haemophilus parainfluenzae', 'Prevotella nanceinsis', 'Neisseria ASV9', 'Veillonella ASV5', 'Streptococcus ASV8', 'Prevotella histcola', 'Prevotella salivae', 'Prevotella melaninogenica', 'Streptococcus mutans', 'Fusobacterium nucleatum_subsp._animalis', 'Veilonella dispar', imp_top%>%pull(ScientificName), hellinger_network$intraStats%>%group_by(module)%>%slice_max(n=2, order_by=kWithin)%>%pull(ScientificName), by_cohra)
#load extra fonts
extrafont::loadfonts()
#abundance info
my_df=ps_long%>% group_by(OTU,CaseEver)%>%dplyr::summarise(m=mean(Abundance))%>%pivot_wider(names_from = CaseEver, values_from=m)%>%dplyr::mutate(as=ifelse(Case>Control, 'Higher mean abundance case', 'Higher mean abundance controls'), m=ifelse(Case>Control, Case, Control))%>%dplyr::mutate('name'=OTU)
#source tidygraph object creation from network object creation
nw<-hellinger_network
source(file.path(code.dir, 'Analysis', 'createTidyGraph.R'))

#plot networks 
network_plot=gg_whole%>%activate(nodes) %>%dplyr::mutate(ScientificNameLabel=ifelse(name %in% imp_top$ASV, paste(ScientificNameLabel, '*'), ScientificNameLabel)) %>%filter(module %in% c('brown', 'yellow'))%>%dplyr::mutate(module=factor(module, levels=c('yellow', 'brown')))%>%ggraph('nicely')+geom_edge_link(aes(width=weight), alpha=0.8, color='grey')+geom_node_point(aes(color=GenusLabel, size=m, shape=factor(as)))+geom_node_text(aes(label=ScientificNameLabel), fontface='italic', family='Times', repel=TRUE, size=3)+guides(size=FALSE, edge_color=FALSE, color=FALSE, edge_width=FALSE)+scale_color_manual('', values=genus_colors)+scale_shape_manual('', values=c('triangle', 'square'))+theme_graph(base_family='sans')+theme(legend.position = 'bottom')+scale_edge_width(range = c(0.1, 2))+facet_nodes(~module, scales='free')+scale_size(range=c(2, 8))+theme(strip.text.x = element_blank())+theme(text=element_text(face='italic'))

network_n=hellinger_network$module.data%>%filter(moduleColor %in% c('brown', 'yellow') & Visit<=IncidentVisit2)%>%group_by(moduleColor, Visit, Case)%>%dplyr::summarise(n=n())%>%pivot_wider(names_from=Case, values_from=n)%>%dplyr::mutate(modAbund=1, label=paste0('Case n=', Case, '\n Control n=', Control), Case='case')%>%filter(Visit %in% c(3, 5, 7, 8, 9, 10))%>%dplyr::mutate(Visit=as.factor(Visit))
#plot change over time 
network_overtime=ggplot(hellinger_network$module.data%>%filter(Visit %in% c(3, 5, 7, 8, 9, 10) & moduleColor %in% c("brown", 'yellow') & Visit<=IncidentVisit2), aes(x=as.factor(Visit), y=modAbund, color=as.factor(Case)))+stat_summary(geom='point', fun.y=mean)+facet_wrap(module_label3~., ncol=2, scales='free_y')+
    stat_summary(fun.data = mean_cl_boot, geom='errorbar', width=0.5)+#stat_compare_means(aes(label = ..p.signif..), size=5, hide.ns = TRUE, label.y=0.50)+
    geom_line(aes(group=UniquePersonID), alpha=0.1)+geom_smooth(aes(group=as.factor(Case), color=as.factor(Case)), se=F)+scale_color_manual(values=c("darkorange3", "royalblue3"), labels=c("ECC case", "Control"), name="")+theme_classic()+
    scale_x_discrete(name="Age (months)", labels=visit_labels2, expand=c(.1, .1))+ylab("Summed module relative abundance")+
    theme(text=element_text(size=14))+theme(axis.text.x=element_text(angle = 45, vjust = 0.5, hjust=.5))+
    #scale_y_continuous(breaks=c(0, .5, 1), minor_breaks = c(.25, .75))+
    theme(legend.position='bottom')+theme(strip.text = element_text(face='italic'))

fig2b=plot_grid(network_overtime, network_plot, ncol=1, labels=c('B', 'C'))
```


## Communities identified through unsupervised clustering are reproducible in an external cohort

```{r network and cst replicability}
prj_files<-c(list.files(file.path(data.out, 'PRJ', 'Network'), full.names = T), list.files(file.path(data.out, 'PRJ', 'CST'), full.names = T)[-4])
prj=new.env()
prj_files%>%purrr::map(function(file){load(file, envir = prj)})

load(file.path(data.out, 'PRJ', 'phyloseq', 'phyasv_postTax.Rdata'), envir = prj)
prj$ps_long=microbiome::transform(prj$prj.filter, 'compositional')%>%psmelt()%>%dplyr::mutate(ScientificName=paste(Genus, ifelse(is.na(Species), OTU, Species), sep=' '))

cohra_names=hellinger_network$intraStats%>%select(module, ScientificName)%>%filter(!str_detect(ScientificName, 'ASV'))%>%group_by(module)%>%nest()

prj_names=prj$hellinger_network$intraStats%>%select(module, ScientificName)%>%filter(!str_detect(ScientificName, 'ASV'))%>%group_by(module)%>%nest()

shared_taxa=data.frame('prj'=character(), 'cohra'=character(), 'common'=character(), 'different'=character(), 
                       n_common=integer(), n_different=integer(), n_cohra=integer(), n_prj=integer())
for( i in 1:length(cohra_names$data)){
  for(j in 1: length(prj_names$data)){
    common=intersect(prj_names$data[[j]]$ScientificName, cohra_names$data[[i]]$ScientificName)
    n_common=length(common)
    different=setdiff(prj_names$data[[j]]$ScientificName, cohra_names$data[[i]]$ScientificName)
    n_diff=length(different)
    n_cohra=nrow(cohra_names$data[[i]]); n_prj=nrow(prj_names$data[[j]])
    new_row=data.frame(prj_names$module[[j]], cohra_names$module[[i]], paste(common, collapse='; '), 
              paste(different, collapse='; '), n_common, n_diff, n_cohra, n_prj)
    names(new_row)=names(shared_taxa)
    shared_taxa=rbind(shared_taxa, new_row)
  }
}
shared_taxa=shared_taxa%>%dplyr::mutate(p_sharedC=n_common/n_cohra, p_sharedP=n_common/n_prj)

bVSb=shared_taxa %>%filter(cohra=='brown'& prj=='brown')
yVSt=shared_taxa %>%filter(cohra=='yellow'& prj=='turquoise')
```


```{r fig 2d}
prj$ps_long=microbiome::transform(prj$prj.filter, 'compositional')%>%psmelt()%>%mutate(ScientificName=paste(Genus, ifelse(is.na(Species), OTU, Species), sep=' '))

nw<-prj$hellinger_network
my_df=prj$ps_long%>% group_by(OTU)%>%dplyr::summarise(m=mean(Abundance))%>%mutate(name=OTU)
source(file.path(code.dir, 'Analysis', 'createTidyGraph.R'))

matching_networks<-gg_whole%>%
  activate(edges)%>%filter(weight>0.03)%>%
  activate(nodes)%>%
  filter(module %in% c('turquoise', 'brown'))%>%
  mutate(module=factor(module, levels=c('turquoise', 'brown'))) %>%
  filter(!node_is_isolated())%>%
  ggraph('nicely')+
  geom_edge_link(aes(width=weight, alpha=weight), alpha=0.8, color='grey')+
  geom_node_point(aes(color=GenusLabel, size=m))+
  geom_node_text(aes(
    label=ifelse(m>0.01 & !str_detect(ScientificNameLabel, 'ASV'), ScientificNameLabel, 
                 ifelse(ScientificName %in% c(by_cohra, "Veillonella dispar"), ScientificNameLabel, ''))),
    fontface='italic', family='Times', 
    repel=TRUE, size=3, max.overlaps=Inf, 
    xlim = c(-Inf, NA), ylim = c(-Inf, Inf))+
  guides(size=FALSE, edge_color=FALSE, color=guide_legend(nrow=2,byrow=TRUE), edge_width=FALSE)+
  scale_color_manual('', values=genus_colors)+
  theme_graph(base_family='sans')+theme(legend.position = 'bottom')+
  scale_edge_width(range = c(0.1, 2))+
  scale_size_continuous(range = c(.5, 3)) +
  theme(strip.text.x = element_blank())+
  theme(text=element_text(face='italic'))+
  facet_nodes(~module_label2, scales='free')+
  coord_cartesian(clip = "off")

```


```{r figure 3}

fig3=plot_grid(network_overtime, network_plot, matching_networks, ncol=1, labels=c('A', 'B', 'C'), rel_heights=c(.3, .3, .4))



cairo_ps(file.path(main.out, 'Figure4.eps'), family='Arial', width=12, height=16)
fig3
dev.off()
cairo_pdf(file.path(main.out, 'Figure4.pdf'), family='Arial', width=12, height=16)
fig3
dev.off()
```

To examine the reproducibility of these identified groupings of salivary bacteria, we subjected publicly available data from a longitudinal cohort of similarly aged children with 16S sequenced saliva samples to the same analytic pipeline [Holgerson *et al.*; PRJ]. 

In the Holgerson *et al.* sample, `r prj$hellinger_network$moduleNum` network modules were identified (Supplemental Table XX). Among these was a network module including among it's most abundant members *Neisseria flavescens*, *Neisseria perflava*, and *Haemophilus parainfluenzae*, with central taxa *Fusobacterium periodonticum*. This was the network module with the most most member ASVs, and included many *Haemophilus*, *Neisseria* and *Streptococcus* ASVs.  `r yVSt$n_common` of the `r yVSt$n_cohra`(`r round(yVSt$p_sharedC, 2)*100`%) ASVs identified to the species level in the corresponding *Haemophilus parainfluenzae-Neisseria* ASV9 COHRA2 network were members of this Holgerson *et al.* network. The *Neisseria* ASV9 amplicon from the COHRA cohort closely related to the *Neiseria perflava* from the Holgerson *et al.* cohort  (Supplemental Figure XX). 

A *Veillonella-Streptococcus* network was also identified in the Holgerson *et al.* cohort. Like its counterpart in the COHRA2 cohort, this network included many *Prevotella* and *Actinomyces* ASVs.  `r bVSb$n_common` of the `r bVSb$n_cohra` (`r round(bVSb$p_sharedC, 2)*100`%) ASVs identified to the species level in the corresponding COHRA2 network were also members of this Holgerson *et al.* network. of them were in this PRJ network module. The *Veillonella ASV5* amplicon from the COHRA cohort was closely related to the *Veillonella dispar* amplicon from this Holgerson *et al.* (Supplemental Figure XX). 

```{r modeling data }
####nets ####
network_models=hellinger_network$module.data%>%
  #left_join(MetaVisit)%>%
  left_join(mv_usc)%>%
  mutate(case_numeric=ifelse(Case=='Case', 1, 0),
         IncidentVisit2f=factor(IncidentVisit2, levels=c(8, 7, 5, 9, 10), 
                               labels=c('36mos', '24mos', '12mos', 
                                        '48 mos', '60 mos')),
         modAbund100=modAbund*100)

network_nest=network_models %>% 
   filter(Visit != IncidentVisit2 & !is.na(Delivery))%>%
  group_by(module, moduleColor,
           module_label, module_label2, 
           module_label3,Visit)%>%
  nest()%>%
  filter(Visit %in% c(5, 7))

####csts ####
cst_models=CST1%>%
  #left_join(MetaVisit)%>%
  left_join(mv_usc)%>%
  mutate(case_numeric=ifelse(Case=='Case', 1, 0), 
         IncidentVisit2f=factor(IncidentVisit2, 
                                levels=c(8, 7, 5, 9, 10), 
                                labels=c('36mos', '24mos', '12mos',
                                         '48mos', '60mos')),
         CST.8f=recode_factor(CST.8,
                              `2`='H. parainfluenzae - Neisseria ASV9', 
                              `4`='Streptoccous ASV1 dominated w/ G. elegans', 
                              `6`='Streptococcus ASV8 - Neisseria ASV12', 
                              `1`='Neisseria ASV12 - Veillonella ASV5', 
                              `3`='Streptococcus ASV1 dominated w/ Gemella ASV2',
                              `5`='Gemella ASV2 - H. parainfluenzae-Neisseria ASV9' ))

cst_nest=cst_models %>% 
  filter(Visit != IncidentVisit2 & 
           CST.8f!='Unclassified' & 
           !is.na(Delivery))%>%
  group_by(Visit)%>%
  nest()%>%
  filter(Visit %in% c(5, 7))

```

```{r make regression table function}
make_table=function(x, predictor='net', include=everything(), v=NULL){
  variable_list=list(modAbund100 ~ 'Summed network module relative abundance (ranging from 0 to 100)', 
                BFCurrent ~ 'Breastfeeding status', 
                antibiotics_within_3mos_of_visit~'Child received antibiotics within 3 mos prior to visit', 
                Prim_Tot_Teeth_Present~'Count of primary teeth emerged', 
                Delivery ~ 'Birth delivery mode', 
                Education_HS ~ 'Maternal education at prenatal visit',
                IncidentVisit2f~'Visit of case diagnosis/control matching', 
                Q38i_Juice ~ 'Juice weekly frequency', 
                WipeTeeth~'Brush/wipe teeth')
  if(predictor=='cst'){variable_list[[1]]=CST.8f~'Community state type'}
  if(!is.null(v)){variable_list=v}
  x %>% 
     tbl_regression(
    include=include,
    exponentiate = TRUE, 
    pvalue_fun = ~style_pvalue(.x, digits = 2),
    label= variable_list) %>% 
  add_global_p() %>%
  bold_p(t = 0.05) %>%
    add_n(location='level')%>%
    add_nevent(location='level')
}
```

```{r net models}
net_models=network_nest%>% 
    filter(module %in% c('brown', 'yellow') &
             Visit %in% c(5, 7))%>%
  mutate(model_unadjusted=purrr::map(data, 
                              ~glm(case_numeric~modAbund100, data=.x,
                                  family='binomial')),
         model_adjusted=purrr::map(data,
                            ~glm(case_numeric~modAbund100+
                                  BFCurrent+
                                  antibiotics_within_3mos_of_visit+
                                  Prim_Tot_Teeth_Present+
                                  Delivery+
                                  Education_HS+
                                  IncidentVisit2f, data=.x,
                                family='binomial')),
         model_sensitivity=purrr::map(data,
                            ~glm(case_numeric~modAbund100+
                                  BFCurrent+
                                  antibiotics_within_3mos_of_visit+
                                  Prim_Tot_Teeth_Present+
                                  Delivery+
                                  Education_HS+
                                  IncidentVisit2f+
                                   Q38i_Juice+
                                   WipeTeeth
                                  , data=.x,
                                family='binomial')))%>%
  mutate(adjusted_tidy=purrr::map(model_adjusted, tidy, 
                           conf.int=T, exponentiate = T), 
         unadjusted_tidy=purrr::map(model_unadjusted, tidy, 
                             conf.int=T, exponentiate = T), 
         sensitivity_tidy=purrr::map(model_sensitivity, tidy, 
                           conf.int=T, exponentiate = T))

my_net_models_short = net_models %>%
  mutate(label=paste0(ifelse(Visit==5, 
                             '12-month visit', 
                             '24-month visit'), ': ',
                      module_label2))%>%
    mutate(adjusted_table=map2(model_adjusted,
                               label,
                               ~make_table(.x, 
                                           include=c('modAbund100'),
                                           v=list(modAbund100~.y))),
           unadjusted_table=map2(model_unadjusted, 
                                 label, 
                                 ~make_table(.x,
                                             include=c('modAbund100'),
                                             v=list(modAbund100~.y))), 
           sensitivity_table=map2(model_sensitivity, 
                                 label, 
                                 ~make_table(.x,
                                             include=c('modAbund100'),
                                             v=list(modAbund100~.y))))%>%
  arrange(Visit, moduleColor)

tbl_merge(
  list(tbl_stack(my_net_models_short$unadjusted_table), 
  tbl_stack(my_net_models_short$adjusted_table), 
  tbl_stack(my_net_models_short$sensitivity_table)),
  tab_spanner=c('Unadjusted', 'Adjusted', 'Sensitivity'))

```

```{r cst models}

my_cst_models=cst_nest%>%
  filter(Visit %in% c( 5, 7))%>%
  mutate(data =if(Visit==5) 
    map(data, ~.x %>%mutate(CST.8f=relevel(CST.8f, ref='Gemella ASV2 - H. parainfluenzae-Neisseria ASV9')))
    else map(data, ~.x))%>%
  mutate(model_unadjusted=map(data, 
                              ~glm(case_numeric~CST.8f, data=.x,
                                  family='binomial')),
         model_adjusted=map(data,
                            ~glm(case_numeric~CST.8f+
                                  BFCurrent+
                                  antibiotics_within_3mos_of_visit+
                                  Prim_Tot_Teeth_Present+
                                  Delivery+
                                  Education_HS+
                                  IncidentVisit2f, data=.x,
                                family='binomial')), 
         model_sensitivity=purrr::map(data,
                            ~glm(case_numeric~CST.8f+
                                  BFCurrent+
                                  antibiotics_within_3mos_of_visit+
                                  Prim_Tot_Teeth_Present+
                                  Delivery+
                                  Education_HS+
                                  IncidentVisit2f+
                                   Q38i_Juice+
                                   WipeTeeth
                                  , data=.x,
                                family='binomial')))%>%
  mutate(adjusted_tidy=map(model_adjusted, tidy, 
                           conf.int=T, exponentiate = T), 
         unadjusted_tidy=map(model_unadjusted, tidy, 
                             conf.int=T, exponentiate = T))


my_cst_models_short = my_cst_models %>% 
    filter( Visit %in% c(5, 7))%>%
  mutate(label=paste0(ifelse(Visit==5, '12-month visit', '24-month visit'), ': Community state type'))%>%
    mutate(adjusted_table=map2(model_adjusted, label, ~make_table(.x, include=c('CST.8f'), v=list(CST.8f~.y))),
           unadjusted_table=map2(model_unadjusted, label, ~make_table(.x, include=c('CST.8f'), v=list(CST.8f~.y))), 
           sensitivity_table=map2(model_sensitivity, label, ~make_table(.x, include=c('CST.8f'), v=list(CST.8f~.y))))

cst_gt_short=tbl_merge(
  list(tbl_stack(my_cst_models_short$unadjusted_table), 
       tbl_stack(my_cst_models_short$adjusted_table), 
       tbl_stack(my_cst_models_short$sensitivity_table)), 
  tab_spanner=c('Unadjusted', 'Adjusted', 'Sensitivity'))
```


# ECC-associated guilds associate with concurrent salivary pH and future *S. mutans* prevalence 

```{r diversity and smutans statistics}
div=readRDS(file.path(data.out, 'phyloseq', 'diversityandsmut.Rds'))
```

```{r Table1}
network_data=hellinger_network$module.data%>%ungroup()%>%unique()%>%
  select(-module_label3, -module, -CentralTaxa1, -CentralTaxa2,
         -module_label, -TopTaxa1,-TopTaxa2,-moduleColor)%>%
  pivot_wider(names_from=module_label2, values_from=modAbund)

table1=left_join(mv_usc, left_join(network_data, 
                               left_join(main_cst, cst_recodes)))%>%
                  left_join(div%>%
              dplyr::select(FoxCavID, Chao1, Shannon,
                            Abundance_Streptococcus_mutans,
                            Prevalence_Streptococcus_mutans, 
                            Abundance_Scardovia_wiggsiae, 
                            Prevalence_Scardovia_wiggsiae))

table1.noincident=table1 %>%filter(Visit < IncidentVisit2)
table2.incident=table1 %>% filter(Visit==IncidentVisit2)
```

```{table 1, eval=F }
theme_gtsummary_compact()
gt_table1=table1.noincident%>%
  filter(Visit %in% c(3,5 ,7))%>%
  mutate(Visit=factor(Visit, levels=c(3, 5, 7), 
                      labels=c('~2-month visit', '~12-month visit',
                                '~24-month visit')))%>%
  select(Visit, Case,  BabyRace, BabySex, Site,
         BFCurrent, antibiotics_within_3mos_of_visit, 
         Prim_Tot_Teeth_Present, Delivery, 
         Education_HS, Q38i_Juice, WipeTeeth))%>%
  tbl_strata(strata=Visit,
             .tbl_fun=~.x %>% 
               tbl_summary(by=Case,
                           type = list(Prim_Tot_Teeth_Present ~ "continuous"), 
                           digits = all_continuous() ~ 1,
                           label = list(BabySex~"Child's sex", 
                                        BabyRace~"Child's race", 
                                        Education_HS~
                                          'Maternal education reported at prenatal visit', 
                                        Delivery~"Delivery", 
                                        Prim_Tot_Teeth_Present~'
                                        Count of primary teeth erupted',  
                                        antibiotics_within_3mos_of_visit~
                                          'Maternal report of child antibiotics within 3 mos prior to visit',
                                        BFCurrent~'Currently breastfed'),
                           statistic = list(all_continuous() ~ "{mean} ({sd})"))%>%
               add_p())


gt_table2=table1.noincident%>%
  filter(Visit %in% c(3,5 ,7))%>%
  mutate(Visit=factor(Visit, levels=c(3, 5, 7), 
                      labels=c('~2-month visit', '~12-month visit',
                                '~24-month visit')))%>%
  select(Visit, Case, Shannon, Chao1, 
         Abundance_Streptococcus_mutans,
         Prevalence_Streptococcus_mutans,
         Abundance_Scardovia_wiggsiae,
         Prevalence_Scardovia_wiggsiae))%>%
  tbl_strata(strata=Visit,
             .tbl_fun=~.x %>% 
               tbl_summary(by=Case,
                           type = list(Prim_Tot_Teeth_Present ~ "continuous",
                                      Abundance_Streptococcus_mutans~'continuous',
                                      Abundance_Scardovia_wiggsiae~'continuous', 
                                    Prevalence_Streptococcus_mutans~'categorical',
                                    Prevalence_Scardovia_wiggsiae~'categorical'), 
                           digits = all_continuous() ~ 1,
                           label = list(
                          Abundance_Streptococcus_mutans~'S. mutans abundance', 
                        Prevalence_Streptococcus_mutans~'S. mutans ASV detected', 
                        Abundance_Scardovia_wiggsiae~'S. wiggsiae abundance', 
                        Prevalence_Scardovia_wiggsiae~'S. wiggsiae ASV detected'),
                           statistic = list(all_continuous() ~ 
                                              "{mean} ({sd})"))%>%
               add_p())



```

```{r}
incident_table=table2.incident%>%
    filter(Visit %in% c(5, 7, 8, 9, 10))%>%
    mutate(Visit=factor(Visit, levels=c(5, 7, 8, 9, 10), 
                        labels=c('~12-month visit',
                                 '~24-month visit', '36-month visit', '48-month visit', '60 month visit')))%>%
    select(Visit, Case, Shannon, Chao1, 
           Abundance_Streptococcus_mutans,
           Prevalence_Streptococcus_mutans,
           Abundance_Scardovia_wiggsiae,
           Prevalence_Scardovia_wiggsiae, BabyRace, BabySex, Site,
           BFCurrent, antibiotics_within_3mos_of_visit, 
           Prim_Tot_Teeth_Present, Delivery, 
           Education_HS)%>%
    tbl_strata(strata=Visit,
               .tbl_fun=~.x %>% 
                   tbl_summary(by=Case,
                               type = list(Shannon ~ 'continuous', Chao1~'continuous', Prim_Tot_Teeth_Present ~ "continuous",
                                           Abundance_Streptococcus_mutans~'continuous', 
                                           Abundance_Scardovia_wiggsiae~'continuous', 
                                           Prevalence_Streptococcus_mutans~'categorical', 
                                           Prevalence_Scardovia_wiggsiae~'categorical'), 
                               digits = all_continuous() ~ 1,
                               label = list(BabySex~"Child's sex", 
                                            BabyRace~"Child's race", 
                                            Education_HS~'Maternal education reported at prenatal visit', 
                                            Delivery~"Delivery", 
                                            Prim_Tot_Teeth_Present~'Count of primary teeth erupted',  
                                            antibiotics_within_3mos_of_visit~'Maternal report of child antibiotics within 3 mos prior to visit',
                                            BFCurrent~'Currently breastfed',
                                            Abundance_Streptococcus_mutans~'S. mutans abundance', 
                                            Prevalence_Streptococcus_mutans~'S. mutans ASV detected', 
                                            Abundance_Scardovia_wiggsiae~'S. wiggsiae abundance', 
                                            Prevalence_Scardovia_wiggsiae~'S. wiggsiae ASV detected'),
                               statistic = list(all_continuous() ~ "{mean} ({sd})"))%>%
                   add_p())
```

```{r salivary pH}
#########create data #######
#join together information on networks, CST, smutans, and diversity
netcst=left_join(left_join(hellinger_network$module.data, 
                               left_join(main_cst, cst_recodes)), MV)%>%
                  left_join(div%>%
                            dplyr::select(FoxCavID, Chao1, Shannon, 
                                        Abundance_Streptococcus_mutans,
                                        Prevalence_Streptococcus_mutans))%>%
  #create cst group variable for plotting
        dplyr::mutate(CSTGroup=case_when(CST.8%in% c(1, 2)~
                                           'Late childhood CSTs',
                                         CST.8 %in% c(3, 4)~
                                           "Infant CSTs", 
                                         CST.8 %in% c(5, 6)~
                                           'Early childhood CSTs'))
#filter to just visits of interest
netcst.57=netcst%>%
  filter(Visit %in% c(5, 7))

#########networks #######
#plot salivary ph by module abundance
net_ph<-netcst.57 %>%
  filter(moduleColor %in% c('yellow','brown'))%>%
  ggplot(aes(y=SalivapH, x=modAbund))+
  geom_point()+
  facet_grid(Visit~module_label3,
             labeller=labeller(Visit=visit_labels2, 
                               module_label3=label_wrap_gen(30)))+
  geom_smooth(color='red', method='lm')+
  theme_classic()+
  stat_cor()+
  theme(strip.text.x =
          element_text(face='italic'))+
  ylab('Salivary pH')+
  xlab('Summed module relative abundance')+
  theme(text=element_text(size=14))

#manuscript - numbers for missing saliva pH 
netcst.57%>%
filter(moduleColor=='brown')%>%
  group_by(Visit, is.na(CSTlabels), is.na(SalivapH))%>%
  summarise(n=n())

#manuscript - correlation for saliva pH by module abundance
cor_fun<-function(df, var1, var2){
  cor.test(df[[var1]], df[[var2]], method='pearson')%>%tidy()
}

hellinger_nest<-group_by(netcst.57, Visit, module_label3)%>%
  nest() %>%
  dplyr::mutate(Saliva_pH=purrr::map(data, cor_fun, 
                                     var1='SalivapH', var2='modAbund'),
                Primary_Teeth=purrr::map(data, cor_fun,
                                         var1='Prim_Tot_Teeth_Present',
                                         var2='modAbund'), 
                N=purrr::map(data, ~.x %>%
                            group_by(is.na(SalivapH))%>%
                            summarise(n=n())))%>%
  unnest(cols=c(Saliva_pH, Primary_Teeth), names_sep='.')

#########csts #######
#cst colors for plotting
cst_colors=c("thistle","lightcoral", "deepskyblue",  "darkorange",
             "royalblue2", "darkorange3", "royalblue4", "darkorange4",
             'grey')
#cst comparisons for statistical testing
cst_compare=list(c("Gemella ASV2 - H. parainfluenzae - Neisseria ASV9",
                   "Streptococcus ASV8 - Neisseria ASV12"),
                 c("H. parainfluenzae - Neisseria ASV9",
                   "Neisseria ASV12 - Veillonella ASV5"), 
                 c("Streptococcus ASV1 dominated w/ Gemella ASV2",
                   "Streptoccous ASV1 dominated w/ G. elegans" ))

#plot cst by salivary ph
cstph=netcst.57%>%
  filter(moduleColor=='brown' & !is.na(CSTlabels))%>%
  ggplot(aes(x=CSTlabels, y=SalivapH, color=CSTlabels))+
  geom_boxplot()+
  geom_jitter(alpha=0.5)+
  facet_grid(Visit~., labeller=labeller(Visit=visit_labels2))+
  stat_compare_means(label.y=7.8)+
  theme_classic()+
  theme(axis.text.x = element_blank())+
  theme(legend.position = 'bottom', 
        legend.text = element_text(face='italic'))+
  scale_color_manual(name="", values=cst_colors,guide=guide_legend(nrow=3))+
  scale_y_continuous(name='Salivary pH', breaks=c(5.5, 6, 6.5, 7, 7.5, 8))+
  scale_x_discrete(name='CST at current visit')+
  theme(legend.position = 'bottom')+
  stat_compare_means(label.y=7.8, label='p.format',comparisons=cst_compare)+
  theme(text=element_text(size=14))

#manuscript - cst salivary ph numbers 
netcst.57%>%
  filter(moduleColor=='brown' & !is.na(CSTlabels))%>%
  group_by(Visit, CSTlabels, is.na(SalivapH))%>%
  summarise(n=n(), m=round(mean(SalivapH), 2))

###########save plot#######
ph_microbes=plot_grid(net_ph, cstph, ncol=2, labels=c('A', 'B'),
                      align = 'h', axis='tb', 
                        rel_widths=c(0.4, 0.6))
ggsave(ph_microbes, file=file.path(main.out, 'Figure5.pdf'), height=8, width=14)

ggsave(ph_microbes, file=file.path(main.out, 'Figure5.eps'), height=8, width=14)

```


```{r future s mutans}
##### create lagged data for s mutans #########
netcst.lag<-netcst %>%
  group_by(UniquePersonID, moduleColor)%>%
  arrange(UniquePersonID, AgeAtExamMonths, moduleColor)%>%
  dplyr::mutate(across(.cols=c(modAbund, SalivapH, 
                               Prevalence_Streptococcus_mutans, 
                               CSTlabels, Dentobuff, Visit), 
                        ~lag(.x, n=1, default=NA, order_by=UniquePersonID), 
                       .names='{.col}.lag'))

##### cst by future smutans #########
cst_lag=plyr::ddply(
  netcst.lag%>% 
    ungroup()%>% 
    filter(((Visit==7 & Visit.lag==5) | (Visit==8 & Visit.lag==7)) &
             !is.na(CSTlabels.lag) &
             moduleColor=='brown'),
  plyr::.(Visit, CSTlabels.lag),
  summarise,
  n=length(Prevalence_Streptococcus_mutans), 
  pos=sum(Prevalence_Streptococcus_mutans == 'Yes'),
  prop = pos / n,
  low = prop.test(pos,n)$conf.int[1],
  high = prop.test(pos,n)$conf.int[2], 
  label=glue::glue('{pos} of {n}'))%>%
  mutate(high=ifelse(prop==0, 0, high))%>%
  ggplot(aes(x=CSTlabels.lag, y=prop,
             ymin=low, ymax=high, fill=CSTlabels.lag, label=label))+
  geom_bar(stat="identity")+
  geom_errorbar()+
  geom_text(aes(x=CSTlabels.lag, y=high+0.04))+
  facet_grid(Visit~., labeller=labeller(Visit=visit_labels2))+
  theme_classic()+
  theme(axis.text.x = element_blank())+
  theme(legend.position = 'bottom', 
        legend.text =element_text(face='italic'))+
  scale_y_continuous('Proportion with S. mutans 16S amplicon detected')+
  scale_x_discrete('CST at previous visit')+
  scale_fill_manual(name="", 
                     values=cst_colors,
                     guide=guide_legend(nrow=3))+
  theme(legend.position = 'bottom')+
  theme(axis.title.y=element_text(face='italic'))+
  theme(text=element_text(size=14))

net_lag<-netcst.lag %>% 
  filter(moduleColor%in%c('brown', 'yellow') & 
           ((Visit.lag==5 & Visit==7) |
            (Visit.lag==7 & Visit==8)))%>% 
  ggplot(aes(x=Prevalence_Streptococcus_mutans, y=modAbund.lag))+
  facet_grid(Visit~module_label2,
             labeller=labeller(Visit=visit_labels2, 
                               module_label2=label_wrap_gen(
                                 width=30)))+
  geom_boxplot()+
  theme_classic()+
  stat_compare_means()+
  ylab("Summed module abundance in the previous visit")+
  xlab("S. mutans 16S amplicon detected in saliva")+
  theme(axis.title.x = element_text(face='italic'))+
  theme(strip.text.x=element_text(face='italic'))+
  theme(text = element_text(size=14))

smut_microbes=plot_grid(net_lag, cst_lag, ncol=2, labels=c('A', 'B'), 
                        align = 'h', axis='tb', 
                        rel_widths=c(0.4, 0.6))

ggsave(smut_microbes, file=file.path(main.out, 'Figure6.pdf'), height=7, width=15)
ggsave(smut_microbes, file=file.path(main.out, 'Figure6.eps'), height=7, width=15)

#manuscript - fishers p  cst 
fishers_cst_test=netcst.lag %>% 
  filter(moduleColor=='brown' & 
           ((Visit==7 & Visit.lag==5) | 
            (Visit==8 & Visit.lag==7)))%>%
  group_by(Visit, Visit.lag)%>%
  nest()%>%
  dplyr::mutate(chi=purrr::map(data, ~xtabs(~Prevalence_Streptococcus_mutans+
                                              CSTlabels.lag, data=.x)%>%
                                 fisher.test()))
#manuscript - cst percent
netcst.lag %>% 
  filter(moduleColor=='brown' & 
           ((Visit==7 & Visit.lag==5) | 
            (Visit==8 & Visit.lag==7)))%>%
  group_by(Visit, Visit.lag, CSTlabels.lag,
           Prevalence_Streptococcus_mutans)%>%
  summarise(n=n())%>%
   group_by(Visit, Visit.lag, CSTlabels.lag)%>%
  mutate(n_tot=sum(n), percent=n/n_tot*100)


#####old stuff #####

epiround<-function(x, n){
  base=1*10^(-n)
  base_char=paste0('<', base)
  ifelse(x<base, base_char, round(x, n))
}
pullRho<-function(name, var, time){
  est=paste0(var, '.estimate')
  pval=paste0(var, '.p.value')
  rho=hellinger_nest %>% filter(str_detect(module_label3, name) & Visit==time)%>%pull(!!est)%>%round(2)
  p=hellinger_nest %>% filter(str_detect(module_label3, name) & Visit==time)%>%pull(!!pval)%>%epiround(3)
  return(paste0('Rho=', rho, '; p-val=', p))
}




```

The groups of co-occurring taxa identified from the unsupervised clustering methods associated with concurrent salivary pH at the 12- and 24-month visits. Increasing abundance of the *H. parainfluenzae & Neisseria ASV9* module was correlated with increasing salivary pH (`r pullRho('Haemo', 'Saliva_pH', 5)`, `r pullRho('Haemo', 'Saliva_pH', 7)` respectively). Increasing abundance of the *Streptococcus ASV1 & Neisseria ASV12 network* was correlated with decreasing salivary pH (`r pullRho('ASV1', 'Saliva_pH', 5)`, `r pullRho('ASV1', 'Saliva_pH', 7)`). A similar trend was apparent when comparing mean salivary pH between samples assigned to the different community state types (Supplemental Figure XX). 

Children who went on to have *S. mutans* amplicon detected in their saliva at the next visit had lower abundances of the *H. parainfluenzae & Neisseria ASV9* network and higher abundances of the *Veillonella ASV5 & Streptococcus ASV8 network* network than children who did not go on to have *S. mutans detected* (Supplementary Figure XX). A similar trend existed for community state types. Of the children assigned to the  *Gemella ASV2 - H. parainfluenzae - Neisseria ASV9* at the 12-month visit, `r smut_lag_cst %>% filter(Visit==7 & CST.lag=='Gemella ASV2 - H. parainfluenzae - Neisseria ASV9' & smutPrevalence=='Yes')%>%pull(percent)`% of children had detectable *S. mutans* amplicon at the 24-month visit, vs `r smut_lag_cst %>% filter(Visit==7 & CST.lag=='Streptococcus ASV8 - Neisseria ASV12' & smutPrevalence=='Yes')%>%pull(percent)`% of the children assigned to the  *Streptococcus ASV8 - Neisseria ASV12* community state type (*P* value from Fishers exact test=`r epiround(v5_cst_smut_fisher$p.value, 2)`). Of the children assigned to the  *H. parainfluenzae - Neisseria ASV9* CST at the 24-month visit, `r smut_lag_cst %>% filter(Visit==8 & CST.lag=='H. parainfluenzae - Neisseria ASV9' & smutPrevalence=='Yes')%>%pull(percent)`% of children had detectable *S. mutans* amplicon at the 36-month visit, vs `r smut_lag_cst %>% filter(Visit==8 & CST.lag=='Neisseria ASV12 - V. dispar' & smutPrevalence=='Yes')%>%pull(percent)`% of children assigned to the *Neisseria ASV12 - Veillonella ASV5* (*P* value from Fisher exact test `r epiround(v7_cst_smut_fisher$p.value, 2)`).

The trajectory of *S. mutans* over time was correlated with changes in important taxa from the unsupervised clustering techniques adn changes in the oral environment. For example, *F. periodonticum*, the central taxa of the *Neisseria-H. parainfluenzae* network, increased in abundance until 12-months of age (Figure 3A). *S. mutans* abundance only rose after 12-months, as did *S. sanguinis* abundance (Figure 3B). This may reflect the preference of *S. mutans* and *S. sanguinis* for hard surfaces: many children become dentulous at and after 9-months of age. The case-associated *S. mutans* abundance increase was also correlated over time with emergence of *Fusobacterium nucleatum subsp animalis*, a member of the same network cluster (Figure 3; Supplemental Figure XX). 

# ECC-associated guilds do not associate with tooth number or age at first tooth emergence

The *H. parainfluenzae & Neisseria ASV9* module did not correlate with the total number of primary teeth present at the 12- or 24-month visits, although the *Streptococcus ASV1 and Neisseria ASV12 module* did (Supplemental Figure XX). Similarly, although the mean number of primary teeth increased from CSTs that were dominant at different ages, the number of primary teeth did not significantly differ between the two dominant CSTs wihtin any one age (Supplemental Figure XX). Among the subset of children recruited from Pennsylvania, data on the approximate age at first tooth emergence was available. In this subset, age at first tooth emergence was not associated with either WCN modules or CSTs.



# Whole-genome shotgun metagenomics of incident samples reveals taxonomic and functional differences between incident case- and control-samples

In a subsample of incident samples from 15 cases presenting with enamel lesions at or after the 36-month visit and 15 corresponding controls, significant differences existed in the abundance of taxonomic groups by case status (Supplemental Figure XX). Notably, *C. albicans*, *Scardovia wiggsiae* and *Streptococcus mutans* were all more abundant in case plaque, *S. mutans* and *Scardovia wiggsiae* were also more abundant in case saliva. *Prevotella salivae*, *Prevotella histicola*, and *Streptococcus salivarius* were more abundant in cases (P_BH<0.05). Neisseria mucosa (P_BH<0.05), Neisseria lactamic (P_BH<0.05), and Neisseria perflava (P_BH<0.1) were more abundant in controls samples.   


```{r volcano plot functions}
#load data
load(file.path(data.out, 'MS', 'deseqresults.Rdata'))
load(file.path(data.out, 'MS', 'taxpathannotation.Rdata'))

cbPalette_volc=c('red', "#E69F00", "#56B4E9", "#009E73", "darkblue", "#0072B2", "purple", "#CC79A7", "#00FF00", 'darkred', "#000000")
taxa_of_interest=c('Neisseria perflava', 'Neisseria mucosa', 'Haemophilus parainfluenzae', 'Fusobacterium periodonticum', 'Fusobacterium nucleatum', 'Streptococcus sangunis', 'Prevotella salivae', 'Actinomyces graevenitzii', 'Streptococcus vestibularis', 'Prevotella histicola', 'Streptococcus sobrinus',
                   'Lachnoanaerobaculum orale', 
                   'Veillonella dispar', 
                   'Veillonella parvula')
t3=t3%>%
  mutate(label=ifelse(Taxa %in% taxa_of_interest & padj<0.05, Taxa,
                      ifelse((-log10(pvalue)>10)|
                               (-log10(pvalue)>10 & SampleType=='saliva') |
                               (SampleType=='saliva' & log2FoldChange < -2),
                             Taxa, '')))%>%
  mutate(xax_lab='log2(FoldChange)')

taxa_volc=t3%>%
  filter(Taxa!='Unmapped')%>%
  ggplot(aes(x=log2FoldChange, y=-log10(pvalue),
             label=label,color=ColorCode))+
    geom_point()+
    geom_text_repel(max.overlaps = 500)+
    facet_grid(xax_lab~SampleType, scales='free_x')+
    scale_color_manual('Significant genera', values=cbPalette_volc, na.translate=FALSE)+
    theme_classic()+
    xlab('\n log2 Fold change')+
    theme(axis.line.x = element_line(arrow = 
                                         grid::arrow(length = unit(0.3, "cm"), ends = "both")))+
    theme(legend.position='bottom')

ggsave(taxa_volc, filename=file.path(main.out, 'Figure7.pdf'), units=c('in'), width=18, height=8)
ggsave(taxa_volc, filename=file.path(main.out, 'Figure7.eps'), units=c('in'), width=18, height=8)



cbPalette_volc=cbPalette_bar=c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "purple", "#CC79A7", "#00FF00", "#000000")

names(cbPalette_volc)=c(r3%>%filter(!is.na(PathAnnotation_01_1) & padj<0.05)%>%pull(ColorCode)%>%unique(), 'padj>0.05')
names(cbPalette_bar)=r3%>%filter(!is.na(PathAnnotation_01_1) & padj<0.05)%>%pull(PathAnnotation_01_1)%>%unique()

annotate_npc <- function(label, x, y, ...)
{
  ggplot2::annotation_custom(grid::textGrob(
    x = unit(x, "npc"), y = unit(y, "npc"), label = label, ...))
}

# Volcano
r3=r3%>%left_join(sig_taxa)%>%dplyr::mutate(label=ifelse(log2FoldChange>15, 
                           paste0(as.character(kname), '\n', '(', Taxa, ')'), ''))
volc=ggplot(r3%>%filter(!is.na(PathAnnotation_01_1)),
            aes(x=log2FoldChange, y=-log10(pvalue),label=label,
              color=ColorCode))+geom_point()+geom_text_repel(size=3)+
        facet_wrap(~SampleType, scales='free_x')+
  scale_color_manual('Annotation', values=cbPalette_volc, na.translate=FALSE)+
  theme_classic()+xlab('\n log2 Fold change ')+
  theme(axis.line.x = element_line(arrow = 
        grid::arrow(length = unit(0.3, "cm"), ends = "both")))+
  theme(legend.position='bottom')+
  guides(color=guide_legend(nrow=3,byrow=TRUE))

#make a dummy layer to provide different x axis limits for facets
dummy=data.frame(log2FoldChange=c(-10, 25, -4, 4), pvalue=c(range(r3$pvalue), range(r3$pvalue)), SampleType=c('plaque', 'plaque', 'saliva', 'saliva'), PathAnnotation_01_3=rep('', 4), ColorCode=rep('padj>0.05', 4), label='')

fig4a=volc+annotate_npc('Case enriched', x=0.8, y=0, vjust=3)+annotate_npc('Control enriched', x=0.1, y=0, vjust=3)+coord_cartesian(clip = "off")+geom_blank(data=dummy)

#filter data frame to only those paths we want to graph
#group by annotation and count
r4=r3%>%filter(padj<0.05 & SampleType=='plaque')%>%dplyr::mutate(Enrichment=ifelse(log2FoldChange>0, 'Enriched in cases', 'Enriched in controls'))%>%dplyr::group_by(PathAnnotation_01_1, PathAnnotation_01_3)%>%filter(!is.na(PathAnnotation_01_1))%>%dplyr::mutate(n=n())%>%arrange(PathAnnotation_01_1, n)%>%dplyr::mutate(PathAnnotation_01_3=factor(PathAnnotation_01_3, levels=unique(PathAnnotation_01_3)))%>%dplyr::group_by(Enrichment, .add=T)%>%dplyr::mutate(n_enrich=n())%>%dplyr::select(SampleType, PathAnnotation_01_1, PathAnnotation_01_3, Enrichment, n, n_enrich, kname)%>%unique()

#toggle rect_width to be either outline (2) or fill (200)
rect_width=200 #2

fig4b= ggplot(r4%>%filter(n>1), aes(x=fct_reorder(PathAnnotation_01_3, n), y=n_enrich))+geom_rect(aes(color=PathAnnotation_01_1), alpha=0, size=rect_width, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf)+geom_bar(stat='identity', aes(fill=Enrichment))+coord_flip()+ggforce::facet_col(~PathAnnotation_01_1, scales='free_y', space='free', labeller = label_wrap_gen())+theme_classic()+xlab('')+ylab('Count of KEGG orthologs with padj<0.05')+theme(legend.position='right')+scale_fill_manual(values=c('black', 'grey'))+scale_color_manual('', values=cbPalette_bar, na.translate=FALSE)+guides(color="none")

fig4b_opt2=ggplot(r4, aes(x=fct_reorder(PathAnnotation_01_3, n), y=n_enrich))+geom_rect(aes(color=PathAnnotation_01_1), alpha=0, size=rect_width, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf)+geom_bar(stat='identity', aes(fill=Enrichment))+coord_flip()+ggforce::facet_col(~PathAnnotation_01_1, scales='free_y', space='free')+theme_classic()+xlab('')+ylab('Count of KEGG orthologs with padj<0.05')+theme(legend.position='right')+scale_fill_manual(values=c('black', 'grey'))+scale_color_manual('', values=cbPalette_bar, na.translate=FALSE)+guides(color="none")+  theme(strip.text.x = element_text(margin = ggplot2::margin(0.1,0,0.1,0, "cm")), axis.text.y=element_text(size=7))

fig4=plot_grid(fig4a, fig4b, rel_widths=c(0.6, 0.4))
fig4_opt2=plot_grid(fig4a, fig4b_opt2, rel_widths=c(0.6, 0.4)) 

ggsave(fig4, filename=file.path(main.out, 'Figure8.pdf'), units=c('in'), width=18, height=8)
ggsave(fig4_opt2, filename=file.path(main.out, 'Figure8_opt2.pdf'), units=c('in'), width=18, height=10)

cairo_ps(file.path(main.out, 'Figure8.eps'), family='Arial', width=18, height=10)
fig4
dev.off()

cairo_ps(file.path(main.out, 'Figure8_opt2.eps'), family='Arial', width=18, height=10)
fig4_opt2
dev.off()

r3_table=r3%>%select(SampleType, KEGG, kname, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, PathAnnotation)%>%arrange(padj)
write.xlsx(r3_table, file=file.path(supp.out, 'FunctionsDeseq2.xlsx'), overwrite = T)
```

Significant differences also existed in the abundance of KEGG pathways in both plaque and saliva samples by case status (Figure 3A; Supplemental Tables XX). Among the top 20 most significantly different KEGG orthologs were additional KEGG orthologs with annotations to antibiotic production, antibiotic resistance and quorum sensing (Supplemental Table XX). Many KEGG orthologs significantly enriched in plaque samples annotated to metabolic or environmental information processing, including ABC transporters, transcription factors, and two component systems [Figure 3B]. 

Functional differences sometimes reflect taxonomic differences. For example, all the case enriched KEGG orthologs annotating to oxidative phosphorylation, including two of the most strongly case-enriched KEGG orthologs, were found only in the genus Candida (Supplemental Table XX and Supplemental Figure XX). Candida was only present in case plaque samples. 

```{r}
print('done')
```


