---
title: 'Testing the ecological hypothesis: Longitudinal case-control study of early
  childhood caries and salivary microbiome assembly supplementary section'
author: "Freida Blostein"
date: "`r Sys.Date()`"
geometry: margin=3cm
header-includes:
  - \renewcommand{\topfraction}{.85}
  - \renewcommand{\bottomfraction}{.7}
  - \renewcommand{\textfraction}{.15}
  - \renewcommand{\floatpagefraction}{.66}
  - \setcounter{topnumber}{3}
  - \setcounter{bottomnumber}{3}
  - \setcounter{totalnumber}{4}
output:
  pdf_document: 
    extra_dependencies: ["flafter"]
  html_document: default
  word_document: default
  
---

```{r, echo = FALSE, include=FALSE, warning=FALSE, message=FALSE}
#prj directory
#knitr::opts_chunk$set(fig.pos = "!H", out.extra = "")
prjdir<-gsub('/Code', '', here::here())
data.out<-file.path(prjdir, 'Data')
code.dir<-file.path(prjdir, 'Code')
output.out<-file.path(prjdir, 'Output')
#original file folders - COHRA
og.data<-file.path(prjdir, 'OGData', 'COHRA')
meta.data<-file.path(og.data, 'Metadata')
dada.data<-file.path(og.data, 'DADA2')
wgs.path=file.path(prjdir, 'OGData', 'COHRA', 'WGS')

#results
main.out<-file.path(output.out, 'Main')
supp.out<-file.path(output.out, 'Supplement')
dir.create(main.out)
dir.create(supp.out)

#options
keepDouble=TRUE

#a billion libraries
#data manipulation
library(dplyr)
library(tidyr)
library(purrr)
library(broom)
library(stringr)
library(stringi)
library(forcats)
library(readr)
library(tibble)
library(openxlsx)
library(compareGroups)
library(psych)
library(mclust)
library(DirichletMultinomial)
library(aricode)

#data visualization
library(ggplot2)
library(ggalluvial)
library(ggpubr)
library(ggtree)
library(ggjoy)
library(ggrepel)
library(tidygraph); 
library(ggraph);
library(igraph)
library(ggvenn)
library(cowplot)
library(patchwork)
library(grid)
library(gridExtra)
library(gtable)
library(kableExtra)

#microbiome
library(phyloseq)
library(phangorn)
library(ape)
library(msa)
library(SQMtools)
library(fgsea)
library(KEGGREST)

#knitr options
knitr::opts_chunk$set(echo = FALSE, include=TRUE, warning=FALSE, message=FALSE, results='hide', fig.keep = 'all')

```

```{r read in data, include=F}
#read in phyloseq object
load(file.path(data.out, 'phyloseq', 'phyasv_meta.Rdata'))
ps_long=microbiome::transform(ps_meta_nodups, 'compositional')%>%psmelt()%>%mutate(ScientificName=paste(Genus, ifelse(is.na(Species), OTU, Species), sep=' '))

#extract taxa info
taxa=as.data.frame(ps_meta_nodups@tax_table)%>%mutate(ASV=row.names(.), ScientificName=paste(Genus, ifelse(is.na(Species), ASV, Species), sep=' '))

#metavisit data
load(file.path(data.out, 'Metadata', 'MetaVisit.Rdata'))
if(keepDouble==T){MV=MetaVisit}
if(keepDouble==F){MV=MetaVisit%>%filter(count!='Double; drop')}

#correct incident visits 
MV = MV %>% mutate(IncidentVisit2=IncidentVisit2_new)

#uscur info 
load(file.path(data.out, 'Metadata', 'myuscur.Rdata'))

#merge metavisit and uscur 
mv_usc = MV %>% 
  left_join(my_uscur)

#
my_uscur= my_uscur %>% 
  left_join(MV)
```

```{r load data supervised and unsupervised, include=FALSE}
unsupervised_files<-c(list.files(file.path(data.out, 'Network'), full.names = T), list.files(file.path(data.out, 'CST'), full.names = T)[-1])

unsupervised_files%>%purrr::map(function(file){load(file, envir = .GlobalEnv)})

clr_network$module.data<-left_join(clr_network$module.data, MV)
hellinger_network$module.data<-left_join(hellinger_network$module.data, MV)

#pick most important cst 
main_cst=CST_k6
#merge with metavisit and rename NA as unclassified
CST1<-left_join(main_cst, MV %>% select(FoxCavID, UniquePersonID, UniqueID, BabySubjectID, Site, Case, IncidentVisit2, Visit, count))
levels(CST1$CST.8)<-c(levels(CST1$CST.8), 'Unclassified')
CST1[is.na(CST1$CST.8),'CST.8']<-"Unclassified"

#read in RF data
load(file.path(data.out, 'RF', 'rfResults_hellinger_iVisit_FALSE_duplicates_FALSE.Rdata'))

#wgs
load(file.path(data.out, 'MS', 'kegg_and_taxa_slim.Rdata'))
#load humann3 object
load(file.path(data.out, 'H3', 'Humann3.Rdata'))
```

```{r visit labels, include=FALSE}
#visit labels for plotting
visit_labels2<-c('Birth', '2mos', 'First tooth \n (mean age=9 months)', '12mos', '24mos', '36mos', '48mos', '60mos')
names(visit_labels2)<-c('2', '3', '4', '5', '7', '8', '9', '10')
visit_labelsV2=visit_labels2
names(visit_labelsV2)<-paste0('V', c('3', '5', '7', '8', '9', '10'))
```

```{r name CSTs, include=FALSE}
cst_recodes<-data.frame('CST.8'=as.factor(c(1, 2, 3, 4, 5, 6)), CSTlabels=factor(c('Neisseria ASV12 - Veillonella ASV5', 'H. parainfluenzae - Neisseria ASV9','Streptococcus ASV1 dominated w/ Gemella ASV2', 'Streptoccous ASV1 dominated w/ G. elegans', 'Gemella ASV2 - H. parainfluenzae - Neisseria ASV9', 'Streptococcus ASV8 - Neisseria ASV12'), levels=c('Streptococcus ASV1 dominated w/ Gemella ASV2', 'Streptoccous ASV1 dominated w/ G. elegans','Gemella ASV2 - H. parainfluenzae - Neisseria ASV9', 'Streptococcus ASV8 - Neisseria ASV12','H. parainfluenzae - Neisseria ASV9', 'Neisseria ASV12 - Veillonella ASV5')))
```

# Description of study sample

We used a nested case-control design with incident sampling to sample sequential saliva samples from 99 ECC cases and 90 incidence-visit matched controls from the Center for Oral Health Research in Appalachia 2 cohort (COHRA2), using the 2019 data freeze. Children in COHRA2 are followed from birth until 5 years of age with regularly scheduled study visits in which they both provide oral samples and are examined for dental decay. Children therefore enter the cohort at birth, and all children are at-risk for developing ECC upon entry into the cohort. At each visit in which cases were diagnosed, we selected a similar number of ECC-free controls from the risk-set of all individuals still at risk of ECC, i.e. all those individuals who have not yet been diagnosed with ECC.  Incidence density sampling does not preclude the re-selection of a control as a case at later time points; controls can also be selected as controls more than once. In this analysis, one control became a case at a later age, and one control was selected as a control in two different risk-sets. 

```{r}
library(DiagrammeR)
wrap=function(x, width=60){
  paste(stri_wrap(x, width=width, indent=0, exdent = 0), collapse='\n')
}

exclusion_graph=grViz("
digraph a_nice_graph {

# node definitions with substituted label text
node [fontname = Helvetica,
      fontcolor = darkslategray,
      shape = rectangle, 
      width=4]
a [label = '@@1']
b [label = '@@2']
c [label = '@@3']
d [label = '@@4']
e [label = '@@5']
f [label = '@@6']
g [label = '@@7']


#point nodes for exclusion  
node [shape=point, width=0.01, height=0.01]
   a1
   b1
   c1
   d1
   e1
   f1
   g1

  

    
## DRAW 

# edge definitions with the node IDs
a -> a1 [arrowhead=none]
a1 -> b
b -> b1 [arrowhead=none]
b1 -> c
c -> c1 [arrowhead=none]
c1 -> d
c1 -> e 
c1 -> f 
c1 -> g [headlabel='Shotgun metagenomic sequencing ', labeljust='l', style=dashed]



}

[1]: glue('1131 children in COHRA2')
[2]: glue(wrap('189 children sampled into the incidence density sampled case-control design'))
[3]: glue(wrap('Children with saliva samples with clean, quality controlled 16S amplicon data: 175 children (177 records) at 2-months, 164 children (166 records) at 12-months, 170 children (172 records) at 24-months -- all clean pre- and incident samples included in unsupervised clustering algorithims and shown in Figures 3 and 4'))
[4]: wrap(glue('Pre-incident samples: 158 (160 records) at 12-months, 132 (134 records) at 24-months --- supervised random forest on 12- and 24-month pre-incident samples (Figure 2, only individuals, no duplicated records), main adjusted models of ECC predicted by unsupervised clusters (Table 3 and 4, includes duplicated record for individuals selected at 2 risk sets, excludes 1 individuals with missing delivery mode'))
[5]: wrap(glue('Pre- and incident samples with both clean 16S amplicon data and salivary pH data: n=94 (95 records) at 12-months, n=160 (162 records) at 24-months  (Figure 5, includes duplicated record for individuals selected at 2-risk sets'))
[6]:  wrap(glue('Pre- and incident samples with clean 16S amplicon data at 12-months and 24-months: n=155 (157 records), at 24-months and 36-months: n=114 (116 records),  (Figure 6, includes duplicated record for individuals selected at 2-risk sets'))
[7]:  wrap(glue('30 children with incidence-visit plaque and saliva sequenced using shotgun metagenomics (Figures 7 and 8)'))
")
library(rsvg)
library(DiagrammeRsvg)
```

```{r, eval=F}
file.remove(file.path(supp.out, 'SupplementalFigure1.pdf'))
exclusion_graph %>%
    export_svg %>%
  charToRaw %>% 
  rsvg_pdf(file = file.path(supp.out, 'SupplementalFigure1.pdf'))
```


```{r FIG 1, fig.cap='Flow chart of samples included in each analysis'}
exclusion_graph
```



Additional file 2 shows the distribution of some key characteristics in the children sampled as controls into the nested case control cohort and in the overall disease free cohort at the 2-month visit. Additional files 3 and 4 have additional data on case phenotype presentation.

```{r compare nested case control and entire cohort}
load(file.path(data.out, 'Metadata', 'CohortData.Rdata'))
ever_d1mft=additionalMeta1%>%filter(any_d1mft=='Yes')%>%pull(BabySubjectID)
library(gtsummary)
compare_cohort=additionalMeta1 %>% ungroup()%>%
  mutate(ever_d1mft=case_when(BabySubjectID %in% ever_d1mft | in_ecc_case=='Selected case' ~ 'Ever d1mft',
                   TRUE ~ 'Never d1mft'))%>%
 filter(ever_d1mft=='Never d1mft')%>%
  filter(PhCall %in% c(1))%>%
  mutate(case_when(PhCall==1~'2-month', PhCall==3~'12-month', PhCall==5~'24-month'))%>%
  select(ever_d1mft, in_ecc, BabySex, BabyAgeAtExam, BabyRace,
         Education_HS,  Delivery, 
         Prim_Tot_Teeth_Present, 
        BFCurrent)%>%
 # mutate(PhCall=factor(PhCall, levels=c(1, 3, 5), 
#                     labels=c('~2-month visit', '~12-month visit',
 #                               '~24-month visit')))%>%
  tbl_strata(strata=ever_d1mft, 
             .tbl_fun=~.x %>%
               tbl_summary(by=in_ecc, 
                           type = list(Prim_Tot_Teeth_Present ~ "continuous"),#, 
                                       #Prim_d1ft~ 'continuous', 
                                       #Prim_d1fs~'continuous'), 
                           digits = all_continuous() ~ 0,
                           label = list(BabySex~"Child's sex", 
                                        BabyAgeAtExam~'Child age (months)', 
                                        BabyRace~"Child's race", 
                                        Education_HS~'Maternal education reported at prenatal visit', 
                                        Delivery~"Delivery", 
                                        #any_d1mft~'Any carious lesion, white spot, filled or missing tooth at visit', 
                                        Prim_Tot_Teeth_Present~'Count of primary teeth erupted/present',#, 
                                        #Prim_d1fs~'Count of primary tooth surfaces with carious lesion, white spot or filling', 
                                        #Prim_d1ft~'Count of primary teeth with carious lesion, white spot, or filling', 
                                        BFCurrent~'Currently breastfed'),
                           statistic = list(all_continuous() ~ "{median} ({min}, {max})"))%>%
               add_n()%>%
               add_p()%>%
               add_overall())


additional_file1=compare_cohort %>% 
  as_gt()%>%
  as.data.frame()%>%
  select(var_label, label, test_name_1, n_1, stat_1_1, stat_2_1, p.value_1)

colnames(additional_file1)[5:6]=c('Not in nested case control sample N=905', 
              'Controls selected into case-control sample N=90')

 write.xlsx(additional_file1, file.path(supp.out, 'AdditionalFile2.xlsx'), overwrite=T)
  
 library(flextable)
 
compare_cohort %>% 
  as_flex_table()%>%
  save_as_docx(path=file.path(supp.out, 'AdditionalFile2.docx'))
```

```{r case severity}
cst_12_24= CST1 %>% 
  filter(Visit %in% c(5, 7))%>%
  mutate(CST.8f=recode_factor(CST.8,
                              `2`='H. parainfluenzae - Neisseria ASV9', 
                              `4`='Streptoccous ASV1 dominated w/ G. elegans', 
                              `6`='Streptococcus ASV8 - Neisseria ASV12', 
                              `1`='Neisseria ASV12 - Veillonella ASV5', 
                              `3`='Streptococcus ASV1 dominated w/ Gemella ASV2',
                              `5`='Gemella ASV2 - H. parainfluenzae-Neisseria ASV9' ), 
         CST.visit=case_when(Visit==5~'12-month microbial community state type', Visit==7~'24-month microbial community state type'))%>%
  select(UniquePersonID, CST.visit, CST.8f)

colnames(cst_12_24)=c('UniquePersonID', 'CST.visit', 'CST')

net_wide=hellinger_network$module.data%>%
  filter(moduleColor %in% c('yellow', 'brown') & 
           Visit %in% c(5, 7))%>%ungroup()%>%
  select(UniquePersonID, Visit, module_label2, modAbund)%>%
  pivot_wider(names_from=module_label2, values_from=modAbund)

dmft_severity=MV %>% 
  filter(IncidentVisit2==Visit)%>%
  left_join(cst_12_24)%>%
  filter(!is.na(CST.visit))%>%
  mutate(proportion_d1ft=
           round(Prim_d1ft/Prim_Tot_Teeth_Present*100),
         proportion_d2ft=round(Prim_d2ft/Prim_Tot_Teeth_Present*100), 
         IncidentVisit2=factor(IncidentVisit2, 
                               levels=c(5, 7, 8, 9, 10), 
                               labels=paste0(c('12-', '24-', '36-',
                                               '48-', '60-'),
                                             'month visit')), 
         Age_d1ft=AgeAtExamMonths)%>%
  #mutate(CST=forcats::fct_explicit_na(CST))%>%
  tbl_strata(strata=CST.visit, 
    ~.x %>%droplevels()%>%
      tbl_summary(by=CST, 
              include=c(IncidentVisit2, 
                        Case,
                        Prim_d1ft, 
                        Age_d1ft,
                        proportion_d1ft,
                        Prim_d2ft, 
                        proportion_d2ft,
                        Prim_Tot_Teeth_Present), 
               type = list(Prim_d2ft ~ "continuous", 
                           Prim_d1ft~'continuous', 
                           Prim_Tot_Teeth_Present~'continuous'),
              statistic = list(all_continuous() ~ 
                                 "{median} ({min}, {max})"), 
              label=list(IncidentVisit2='Incident visit', 
                         Case='Case/control', 
                         Age_d1ft='Age (months) at diagnosis',
                         Prim_d1ft='Count of primary teeth with fillings, decay, or white spots (case definition) at incident visit', 
                         Prim_d2ft='Count of primary teeth with fillings or decay (excludes white spots) at incident visit', 
                         Prim_Tot_Teeth_Present='Count of primary teeth present at incident visit', 
                         proportion_d1ft='Percent of present teeth with fillings, decay, or white spots (%) at incident visit',
                         proportion_d2ft='Percent of present teeth with fillings or decay (excludes white spots, %) at incident visit')))

dmft_severity%>% 
  as_flex_table()%>%
  save_as_docx(path=file.path(supp.out, 'AdditionalFile3.docx'))

dmft_severity_individual=MV %>% 
  filter(IncidentVisit2==Visit & Case=='Case')%>%
  left_join(cst_12_24%>% 
             pivot_wider(names_from = CST.visit, values_from = CST))%>%
  left_join(net_wide%>%mutate(Visit=ifelse(Visit==5, '12-month', '24-month'))%>%pivot_wider(names_from=Visit, values_from=contains('network')))%>%
  mutate(proportion_d1ft=
           round(Prim_d1ft/Prim_Tot_Teeth_Present*100),
         proportion_d2ft=round(Prim_d2ft/Prim_Tot_Teeth_Present*100), 
         IncidentVisit2=factor(IncidentVisit2, 
                               levels=c(5, 7, 8, 9, 10), 
                               labels=paste0(c('12-', '24-', '36-',
                                               '48-', '60-'),
                                             'month visit')))%>%
  select(IncidentVisit2, contains('community'), contains('network'),
         Prim_d1ft, Prim_d2ft, Prim_Tot_Teeth_Present)

 write.xlsx(dmft_severity_individual, file.path(supp.out, 'AdditionalFile4.xlsx'), overwrite=T)
```

Among the nested case-control sample, cases and controls did not differ in the distribution of race, sex, mode of birth delivery, or site of recruitment. Cases and controls also did not differ in maternal report of child receipt of antibiotics within the 3 months immediately prior to a visit. At each of the 2-, 12- and 24-month visits, cases and controls had similar numbers of erupted primary teeth. Fewer cases (49%) than controls (63%) were breastfed at the 2-month visit, although this was not a statistically significant difference ($P$=0.07). The mothers of cases were less likely to have an education beyond a high school degree (33%) than the mothers of controls (63%, $P$<0.001). 

All available saliva samples from the cases and controls from the visit in which they were selected and all preceding visits were pulled for 16S rRNA amplicon sequencing. Selected individuals occasionally missed visits, did not have a saliva sample available, or had a saliva sample which failed 16S amplicon quality control (Additional file 5).

```{r TABLE 2, include=FALSE}
source(file.path(code.dir, 'Analysis', 'sampleLoss.R'))
write.xlsx(lostSamples, file.path(supp.out, 'AdditionalFile5.xlsx'), overwrite=T)
```

Additionally, we randomly selected a subcohort of 15 cases presenting with enamel lesions at or after the 36-month visit and 15 corresponding controls. Incident-visit plaque and saliva samples from these 30 individuals were submitted for whole genome shotgun metagenomic sequencing. Since only children who developed dental decay at or after the 36-month visit were eligible to be sampled into the shotgun metagenomic sequencing project, these children are not truly representative of the entire nested case-control study. However, the subsample submitted for metagenomic sequencing did not significantly differ from the overall nested case-control sample in the distribution of race, sex, recruitment site, maternal education, maternal report of antibiotics in the 3-months preceeding the 2-, 12- and 24-month visit, breastfeeding at the 2-, 12-, or 24-month visit (see Additional File 6).

```{r demographics of wgs sample}
wgs_demographics=MetaVisit %>%
  filter(sample_in_wgs ==
           'Sample in shotgun sequenced subset' &
           Visit == IncidentVisit2) %>%
  mutate(Visit=factor(Visit, levels=c(8, 9, 10), 
                      labels=paste0(c('36', '48', '60'), 
                                    '-month visit')))%>%
  select(
    Site,
    Visit,
    BabySex,
    BabyRace,
    Delivery,
    Education_HS,
    SalivapH,
    Prim_d1ft,
    Prim_d2ft,
    Prim_Tot_Teeth_Present,
    Case,
    antibiotics_within_3mos_of_visit,
    BFDurationCat
  ) %>%
  tbl_summary(
    by = Case,
    type =
      list(Prim_Tot_Teeth_Present ~ "continuous",
           Prim_d1ft ~ 'continuous', 
           Prim_d2ft ~ 'continuous',
           Prim_Tot_Teeth_Present~ 'continuous',
          SalivapH ~ 'continuous'),
    statistic = list(all_continuous() ~ "{median} ({min}, {max})"),
    digits = all_continuous() ~ 0,
    label = list(
      BabySex ~ "Child's sex",
      BabyRace ~ "Child's race",
      Education_HS ~ 'Maternal education at prenatal visit',
      Delivery ~ "Delivery",
      Prim_Tot_Teeth_Present ~ 'Count of primary teeth erupted/present',
      Prim_d1ft ~ 'Count of primary teeth with carious lesion, white spot, or filling (case definition)',
      Prim_d2ft ~ 'Count of primary teeth with carious lesion or filling (excludes white spots)',
      Visit ~ 'Visit of case diagnosis',
      antibiotics_within_3mos_of_visit ~ 'Maternal report of child antibiotics within 3 mos prior to visit',
      BFDurationCat ~ 'Duration of breastfeeding'
    )
  )
```

```{r compare wgs subset to nested group}
MV %>% 
  filter(Visit %in% c(3,5, 7))%>%
  mutate(Visit = factor(Visit, levels=names(visit_labels2), labels=visit_labels2), 
         IncidentVisit2=factor(IncidentVisit2,levels=names(visit_labels2)[4:8], 
                               labels=visit_labels2[4:8]))%>%
  select(Site, BabySex, BabyRace, Delivery, Education_HS, SalivapH, 
         Prim_Tot_Teeth_Present, Case,
         antibiotics_within_3mos_of_visit, IncidentVisit2, Visit,
         person_in_wgs, BFCurrent)%>%unique()%>%
  tbl_strata(strata=Visit, 
              .tbl_fun = ~.x %>% 
               tbl_summary(by=person_in_wgs,
                           type = list(Prim_Tot_Teeth_Present ~ "continuous", 
                                       SalivapH ~ 'continuous'), 
                           digits = all_continuous() ~ 0,
                           label = list(BabySex~"Child's sex", 
                                        BabyRace~"Child's race", 
                                        Education_HS~'Maternal education reported at prenatal visit', 
                                        Delivery~"Delivery", 
                                        Prim_Tot_Teeth_Present~'Count of primary teeth erupted/present', 
                                        
                                        IncidentVisit2~'Visit of case diagnosis',
                                        antibiotics_within_3mos_of_visit~'Maternal report of child antibiotics within 3 mos prior to visit',
                                        BFCurrent~'Currently breastfed'),
                           statistic = list(all_continuous() ~ "{median} ({min}, {max})"))%>%
               add_p())%>%
  as_flex_table()%>%
   save_as_docx(path=file.path(supp.out, 'AdditionalFile6.docx'))
```


# 16S amplicon bioinformatics

## Sequencing

Samples were sequenced over 3 runs (each with 4 plates). Each plate included a positive mock control (ZymoBIOMICS Microbial Community DNA Standard (Zymo Research cat# D6306) and 2 negative controls (extraction control and water control). We processed sequenced reads. Sequencing was done on the Illumina MiSeq platform, using a MiSeq Reagent Kit V2 500 cycles (Illumina cat# MS102-2003), according to the manufacturers instructions with modifications found in the Schloss SOP.  Accuprime High Fidelity Taq (Life Technologies cat # 12346094 ) was used instead of Accuprime Pfx supermix. PCR products are visualized using an E-Gel 96 with SYBR Safe DNA Gel Stain, 2% (Life technologies cat# G7208-02).  Libraries are normalized using SequalPrep Normalization Plate Kit (Life technologies cat # A10510-01) following the manufactures protocol for sequential elution.  The concentration of the pooled samples is determined using Kapa Biosystems Library Quantification kit for Illumina platforms (KapaBiosystems KK4824).  The sizes of the amplicons in the library are determined using the Agilent Bioanalyzer High Sensitivity DNA analysis kit (cat# 5067-4626).  The final library consists of equal molar amounts from each of the plates, normalized to the pooled plate at the lowest concentration. 	Library Preparation for Sequencing and Sequencing Libraries are prepared according to Illumina’s protocol for Preparing Libraries for Sequencing on the MiSeq (part# 15039740 Rev. D) PhiX (15%) was added in 16s amplicon sequencing to create diversity.   Sequencing reagents are prepared according to the Schloss SOP, custom read 1, read 2 and index primers are added to the reagent cartridge.  FASTQ files are generated for paired end reads. DNA was quantified using PicoGreen.						

## DADA2 processing

```{r dada2 and preprocess info, include=FALSE}
track_reads=readRDS(file.path(dada.data, 'track_postchimera_HOMD.rds'))
readlimit=1000
load(file.path(output.out, 'Preprocess', paste0('preprocessParamsStats_', readlimit, '.Rdata')))
load(file.path(output.out, 'Preprocess', paste0('filtersamples_', readlimit, '.Rdata')))
```

Reads were processed to amplicon sequence variants (ASVs) using DADA2 (version 1.14.1) and the Human Oral Microbiome Database (HOMD) version 15.2. After examining quality plots for forward and reverse reads, reads were trimmed at the 240th and 200th nucleotide position respectively, or at the first instance of a quality score <= to 11. The following parameters were used for filtering reads: a maximum of 2 expected errors for both forward and reverse reads and no ambiguous bases. 

Errors were learned (independently for each run) using 1 million bases.  The forward and reverse reads were denoised and then forward and reverse reads were merged together. 

At this point, the runs were merged together, and chimeras were removed using DADA2s removeBimeraDenovo commmand and the consensus method. The mean read number, prior to any sample or taxa filtering and including both positive and negative controls, was `r round(mean(track_reads))`, sd=`r round(sd(track_reads))`. A total of m=`r taxa_count$all` ASVs were identified.

Subsequently, reads were assigned taxonomic names to the species level using the assignTaxonomy and the addSpecies commands of DADA2. A modified version of the Human Oral Microbiome Database (HOMD) version 15.2 (modified by adding the exact sequences of the positive control samples) was used. 

## Sample and taxa filtering

To identify potential contaminant taxa we used the R package decontam, identifying contaminants using both the frequency-based method (PicoGreen quantification used as measure of concentration of amplified DNA), and the prevalence based method; thresholds set at `r paste(decontam.thresh, collapse=' and ')` respectively. Contaminant identification was performed independently within each batch. Of the original `r taxa_count$all` ASVs, `r taxa_count$all-taxa_count$decontam` ASVs were identified as contaminants and removed.

Next, we filtered out samples that had less than `r readlimit` reads (Supplemental Figure 2; red line at read limit=`r readlimit`). Of a total original `r sample_count$all['True Sample']` samples, `r sample_count$lessthan1000['True Sample']` samples had < `r readlimit`, reads, leaving `r sample_count$morethan1000['True Sample']` samples. Three of the true samples which passed quality control were missing all metadata and were excluded, leaving a total of `r sample_count$truesamples['True Sample']` in the analysis. After restricting to just the true samples (not controls) with > `r readlimit`, `r taxa_count$truesamples` ASVs remained. Calculation of diversity metrics (Shannon index, Chao1) was performed on these ASVs. ASVs were not collapsed to the genus or species level. 

```{r FIG 2, fig.cap='Read count by sample type in true samples and positive and negative controls'}
cairo_pdf(file.path(supp.out, 'SupplementalFigure2.pdf'), family='Arial', width=13.2, height=8)
readcountsbyType_preDecontam
dev.off()
readcountsbyType_preDecontam
```

Of the original `r sample_count$all['Mock']` mock communities, `r sample_count$lessthan1000['Mock']` had less than `r readlimit ` reads; the other mock communities in the same run on other plates amplified and were as expected (Supplemental Figure 3). None of the extraction controls had more than `r readlimit` reads, and one water control did, this water control had <3000 reads of highly common taxa in the analytic subset and likely reflects some cross-contamination. 

```{r FIG 3, fig.cap="Mock communities by sequencing run as compared to true mock abundance", fig.width=10}
cairo_pdf(file.path(supp.out, 'SupplementalFigure3.pdf'), family='Arial', width=13.2, height=8)
ps_mock_bar
dev.off()
ps_mock_bar
```


Finally, for the unsupervised and supervised analysis, we further restricted the ASV set to ASVs. We filtered out ASVs that were both 1) present in less than `r prev*100`% of the samples and 2) represented less than `r abund.filter*100`% of the relative abundance in samples in which they were present. This left a total of `r taxa_count$taxafilter` ASVs. 


# Supervised random forest

## Additional investigation of *Streptococcus* amplicons identity 

```{r randomForest}
imp<-lapply(rfFits_hellinger[c(1, 3)], function(x) as.data.frame(x$finalModel$importance)%>%mutate(ASV=row.names(x$finalModel$importance)))%>%bind_rows(.id="column_label")%>%mutate(Visit=as.integer(gsub('V', '', gsub('All taxa', '', column_label))), Month=ifelse(Visit=='5', '12 month', '24 month'))
#info on >mean in case | control
greater_mean=ps_long %>% filter(Visit %in% c(5, 7))%>%group_by(OTU, Visit, Case, ScientificName)%>%summarise(mean=mean(Abundance))%>%pivot_wider(id_cols=c(OTU, Visit, ScientificName), names_from=Case, names_prefix='Abundance_', values_from=mean)%>%mutate(GreaterMean=ifelse(Abundance_Case>Abundance_Control, 'Case', 'Control'), ASV=OTU)
#join 
imp=left_join(imp, greater_mean)
imp_top=rbind(imp%>%group_by(Visit)%>%slice_max(n=10, order_by=MeanDecreaseGini), imp%>%filter(str_detect(ScientificName, 'mutans')))
```


```{r streptococcus blast}
mystreps=ps_meta_nodups@refseq[c('ASV8', 'ASV14', 'ASV82')]
Biostrings::writeXStringSet(mystreps, file.path(data.out, 'Reference', 'mystreps.fasta'))
rm(mystreps)
#run blast in web browser or terminal on mystreps.fasta

#load in description tables
files=list.files(path=file.path(data.out, 'Reference', 'BLAST'), pattern="*.csv", full.names=T)
strep_blast=sapply(files, read_csv, simplify=FALSE)%>%bind_rows(.id="id")
strep_blast=strep_blast%>%mutate(asv=gsub('.*/(asv[0-9]{1,2}).*.csv', '\\1', strep_blast$id))%>%select(asv, everything())%>%select(-id)
```

```{r TABLE 3 4 5}
write.xlsx(strep_blast %>% split(.$asv),
           file.path(supp.out, 'AdditionalFile7.xlsx'), overwrite=T)
```

```{r streptococcus correlation plots}
h3_vs_16S=left_join(species_info$species_long, 
                    ps_long%>%
                      select(-Species, -SampleType, -CaseStatus), 
                    by=c('COHRAID', 'Visit'))%>%
            mutate(case_ever_numeric=ifelse(CaseEver=='Case', 1, 0))

ms_species=left_join(as.data.frame(t(saliva_species_percent))%>%rownames_to_column(var='SampleName')%>%pivot_longer(cols=-SampleName, names_to='Species', values_to='WGSAbundance'), species_info$sample_table)

ms_vs_16S=left_join(ms_species, ps_long%>%select(-Species, -SampleType, -CaseStatus), by=c('COHRAID', 'Visit'))%>%mutate(case_ever_numeric=ifelse(CaseEver=='Case', 1, 0))

cor_partial<-function(df){
  #partial spearman correlation to control for case status and visit
  #Resources: 
  #https://towardsdatascience.com/keeping-an-eye-on-confounds-a-walk-through-for-calculating-a-partia  l-correlation-matrix-2ac6b831c5b6
  #http://sherrytowers.com/2013/09/23/correlations-and-partial-correlations/
  #code from psych package, namely: http://personality-project.org/r/html/partial.r.html
  #https://www.rdocumentation.org/packages/psych/versions/2.0.12/topics/corr.test
  #n for sample size in corr.test is n-s where s is the number of variables partialed out, here I      used n-2 (CaseEver & Visit)= 30-2=28
  rho<-corr.p(partial.r(data=df, x=cs(WGSAbundance, Abundance), y=cs(case_ever_numeric, Visit), 
                        method='spearman'), n=28)
  return(data.frame('rho'=rho$r[2], 'pval'=rho$p[2]))}

calculateWGS16Scor=function(df, genus, asv_list=NA){
  if(!is.na(asv_list)){new_df=df %>% filter(str_detect(Species, genus) & OTU %in% asv_list)}
  if(is.na(asv_list)){new_df=df %>% filter(str_detect(Species, genus) & str_detect(Genus, genus))}
  
  grouped_df=group_by(new_df, SampleType, Species, ScientificName)%>%nest()%>%
    mutate(cor=purrr::map(data, cor_partial))%>%unnest(cor)%>%
    mutate(rho_ifsig=ifelse(pval<0.05, round(rho, 2), NA))
  new_plot=ggplot(grouped_df, aes(x=ScientificName, y=Species, fill=rho, label=rho_ifsig))+
    geom_tile()+geom_text()+facet_wrap(~SampleType)+xlab('16S amplicon')+ylab('WGS metagenomics')+
    theme_classic()+
    scale_fill_gradient2(name='Partial Spearman rho', mid="#FBFEF9",low="#0C6291",high="#A63446", 
                         limits=c(-1, 1)) +
    theme(axis.text=element_text(face='italic'))+theme(axis.text.x=element_text(angle=45, hjust=1))
  return(new_plot)
}

#filter to only saliva for strep plot in H3 & remove streps that are only in plaque
strep_nosaliva=h3_vs_16S%>%filter(SampleType=='Saliva')%>%group_by(Species)%>%summarise(sum=sum(WGSAbundance))%>%filter(sum==0)%>%pull(Species)
saliva_only=h3_vs_16S%>%filter(SampleType=='Saliva' & !(Species %in% strep_nosaliva))

#put plots together
strep_MS=calculateWGS16Scor(ms_vs_16S, 'Streptococcus', c('ASV1', 'ASV10', "ASV8", 'ASV14', 'ASV82', 'ASV26', 'ASV76', 'ASV196'))
strep_H3=calculateWGS16Scor(saliva_only, 'Streptococcus', c('ASV1', 'ASV10', "ASV8", 'ASV14', 'ASV82', 'ASV26', 'ASV76', 'ASV196'))
strep_cors=strep_H3/strep_MS+plot_annotation(tag_levels='A')
#save supplemental plot
#ggsave(strep_cors, file=file.path(supp.out, 'SupplementalFigure_strepcors.eps'), width=8, height=10)
#remove large objects to preserve memory
rm(saliva_only, ms_species, h3_species)
```


```{r diversity and smutans statistics}
div=readRDS(file.path(data.out, 'phyloseq', 'diversityandsmut.Rds'))
```

```{r TABLE 6}
network_data=hellinger_network$module.data%>%ungroup()%>%unique()%>%
  select(-module_label3, -module, -CentralTaxa1, -CentralTaxa2,
         -module_label, -TopTaxa1,-TopTaxa2,-moduleColor)%>%
  pivot_wider(names_from=module_label2, values_from=modAbund)

table1=left_join(MV, left_join(network_data, 
                               left_join(main_cst, cst_recodes)))%>%
                  left_join(div%>%
              dplyr::select(FoxCavID, Chao1, Shannon, Abundance_Streptococcus_mutans,
                            Prevalence_Streptococcus_mutans, Prevalence_Scardovia_wiggsiae, Abundance_Scardovia_wiggsiae))

table1.noincident=table1 %>%filter(Visit < IncidentVisit2)
table2.incident=table1 %>% filter(Visit==IncidentVisit2)

res2<-compareGroups(Case~Shannon+Chao1+`Streptococcus ASV1 & Neisseria ASV12 network`+`Haemophilus parainfluenzae & Neisseria ASV9 network`+`Veillonella ASV5 & Streptococcus ASV8 network`+`Veillonella sp._HMT_780 & Granulicatella elegans network`+`Gemella ASV2 & Leptotrichia shahii network`+CSTlabels+Prevalence_Streptococcus_mutans+Abundance_Streptococcus_mutans+Prevalence_Scardovia_wiggsiae+Abundance_Scardovia_wiggsiae,data=table2.incident%>%filter(Visit%in% c( 5, 7, 8)), include.miss=F)
strata.res2=strataTable(createTable(res2), 'Visit')
visits=c(5, 7, 8, 9, 10)
res_list=list()
for(i in 1:length(visits)){
  data=table2.incident%>% filter(Visit==visits[i])
  res=createTable(compareGroups(Case~Shannon+Chao1+`Streptococcus ASV1 & Neisseria ASV12 network`+`Haemophilus parainfluenzae & Neisseria ASV9 network` +
                `Veillonella ASV5 & Streptococcus ASV8 network`+`Veillonella sp._HMT_780 & Granulicatella elegans network`+
                  `Gemella ASV2 & Leptotrichia shahii network`+CSTlabels+Prevalence_Streptococcus_mutans+Abundance_Streptococcus_mutans+Prevalence_Scardovia_wiggsiae+Abundance_Scardovia_wiggsiae,data=data, include.miss=F))
  res_list[i]=res
  #export2xls(res, file=file.path(supp.out, paste0('SupplementalTable2_v', visits[i], '.xlsx')))
}
names(res_list)=paste0(visit_labels2[names(visit_labels2)%in% visits])
write.xlsx(res_list, file=file.path(supp.out, 'AdditionalFile8.xlsx'), overwrite=T, rowNames=T)

```

```{r}
res<-compareGroups(Case~Shannon+Chao1+`Streptococcus ASV1 & Neisseria ASV12 network`+`Haemophilus parainfluenzae & Neisseria ASV9 network`+`Veillonella ASV5 & Streptococcus ASV8 network`+`Veillonella sp._HMT_780 & Granulicatella elegans network`+`Gemella ASV2 & Leptotrichia shahii network`+CSTlabels+Prevalence_Streptococcus_mutans+Abundance_Streptococcus_mutans+Prevalence_Scardovia_wiggsiae+Abundance_Scardovia_wiggsiae,data=table1.noincident%>%filter(Visit%in% c(3, 5, 7)), include.miss=F)
strata.res=strataTable(createTable(res), 'Visit')

#loop over visits of interest and add to xlsx files 
visits=c(3, 5, 7)
res_list=list()
for(i in 1:length(visits)){
  data=table1.noincident%>% filter(Visit==visits[i])
  res=createTable(compareGroups(Case~Shannon+Chao1+`Streptococcus ASV1 & Neisseria ASV12 network`+`Haemophilus parainfluenzae & Neisseria ASV9 network` +
                `Veillonella ASV5 & Streptococcus ASV8 network`+`Veillonella sp._HMT_780 & Granulicatella elegans network`+
                  `Gemella ASV2 & Leptotrichia shahii network`+CSTlabels+Prevalence_Streptococcus_mutans+Abundance_Streptococcus_mutans+Prevalence_Scardovia_wiggsiae+Abundance_Scardovia_wiggsiae,data=data, include.miss=F))
  res_list[i]=res
}
names(res_list)=paste0(visit_labels2[names(visit_labels2)%in% visits])
write.xlsx(res_list, file=file.path(supp.out, 'AdditionalFile9.xlsx'), overwrite=T, rowNames=T)
```


```{r strep asv identity}
asv8_identity=paste(strep_blast %>%filter(asv%in% c('asv8') & `Per. ident`==100)%>%pull(`Scientific Name`)%>%unique(), collapse=' and ')
asv14_identity=paste(strep_blast %>%filter(asv%in% c('asv14') & `Per. ident`==100)%>%pull(`Scientific Name`)%>%unique(), collapse=' and ')
asv82_identity=paste(strep_blast %>%filter(asv%in% c('asv82') & `Per. ident`==100)%>%pull(`Scientific Name`)%>%unique(), collapse=' and ')
```

Species-level identification of *Streptococcus* with the V4 amplicon of the 16S gene is not perfect. Two amplicons identified as *Streptococcus* were important features in the taxa-wide random-forest, but did not identify to the species level. The identity of the amplicon identified as *Streptococcus mutans* is central to our test of the ecological hypothesis. We performed several analyses to improve our confidence in the taxonomic classification of these important *Streptococcus* amplicons. 

*Streptococcus ASV82*, which identified as *S. mutans*, had 100% identity with *`r asv82_identity`* in a BLAST search. 6 of the top 10 top hits were *S mutans*. None of the other 4 top 10 hits were human oral *Streptococcus* species (Additional file 4). 

Neither *Streptoccocus ASV8* nor *Streptococcus ASV14* had a high identity with any *S. mutans* 16S sequences when performing a BLAST search. *Streptococcus ASV8* had 100% identity with *`r asv8_identity`*, *Streptococcus ASV14* with *`r asv14_identity`*. Neither *Streptococcus ASV8* or *ASV14* had an *S. mutans* BLAST hit in the top 100 best blast hits (Additional file 7). 

Additionally, we exploited our subcohort with metagenomically sequenced saliva samples to interrogate the identity of the *Streptococcus* amplicons. Abundance of the 16S amplicon *Streptococcus ASV82* (*S. mutans*) was strongly and significantly correlated with abundances of *S. mutans* in WGS samples when using either Humann3 profile abundances or abundances estimated from an assembly based pipeline, while amplicon abundances of *Streptococcus ASV8* correlated strongly and significantly with *S. salivarius* and *Streptococcus ASV14* with *S. australis* and *S. infantis* (Supplemental Figure 4, Pearson r values significant at *P* value<0.05 shown as text in the heatmap). 

```{r FIG 4, fig.width=8, fig.height=10, fig.cap='Correlation between Streptococcus amplicons from 16S rRNA gene sequencing and taxa identified from contiguous sequences of whole genome shotgun sequening'}
ggsave(strep_cors, file=file.path(supp.out, 'SupplementalFigure4.eps'), width=8, height=10)
strep_cors
```

The prevalence and abundance of *S. mutans* in case and control samples from visits prior to case diagnosis are shown in Main table 1, while the prevalence and abundance of *S. mutans* in case and control samples from the visit of case diagnosis are in additional file 8. 

# Unsupervised clustering techniques

## Community state typing: choice of k community state types

Using Dirichlet multinomial community state typing, we identified 6 community state types (also called enterotypes). We chose k=6 community state types as this choice of k produced the best model fit as measured by the LaPlace metric (Supplemental Figure 5). Each sample is assigned to a single community state type based on a posterior probability; any sample which did not assign to a community state type at a posterior probability >80% was labeled unclassified. Community state types were named by the taxa defining their separation. The distribution of community state types in incident visits is in Additional file 8 and in preincident visits is shown in addition file 9. 


```{r FIG 5, fig.cap='Laplace model fit at k=1:10 community state types'}
load(file.path(og.data, 'DMM_fit_asvtrim.rda'))
### The return value can be queried for measures of fit (Laplace, AIC, BIC); these are plotted for different k
fit<-fit_asvtrimmed
lplc <- sapply(fit, laplace)
plot(lplc, type="b", xlab="Number of Dirichlet Components", ylab="Model Fit")
cairo_pdf(file.path(supp.out, 'SupplementalFigure5.pdf'), family='Arial', width=13.2, height=8)
plot(lplc, type="b", xlab="Number of Dirichlet Components", ylab="Model Fit", main='Laplace model fit at k=1:10 for community state typing')
dev.off()
```

Supplemental Figure 6 shows the relationship between Shannon diversity and community state type. Each dot is a single sample which has been assigned to a community state type (x-axis), the Shannon index in that sample is on the y-axis Community state type is significantly associated with Shannon diversity. As children age, the dominant community state types progress from those that are *Streptococcus* dominated to more diverse community state types. The progression of community state types in both cases and controls follows this pattern. 


```{r FIG 6, fig.cap='Shannon index in samples by visit and community state type', fig.width=13}
fig7=table1%>%ggplot(aes(x=CSTlabels, y=Shannon, color=CSTlabels))+geom_boxplot()+geom_jitter(alpha=0.2)+facet_wrap(~Visit,nrow=1, labeller=labeller(Visit=visit_labels2))+stat_compare_means(label='..p.signif..')+scale_color_manual(name="Community state type", values=c("thistle", "lightcoral", "deepskyblue",   "darkorange", "royalblue2",  "darkorange3",  "royalblue4", "darkorange4", 'grey', 'black'), guide=guide_legend(nrow=2))+theme_classic()+theme(axis.text.x = element_blank())+theme(legend.text=element_text(face='italic'))+theme(legend.position='bottom')+xlab('')#+stat_compare_means(label='p.signif', label.y=4, comparisons=list(c("Gemella ASV2 - H. parainfluenzae - Neisseria ASV9", "Streptococcus ASV8 - Neisseria ASV12"), c("H. parainfluenzae - Neisseria ASV9", "Neisseria ASV12 - V. dispar")))
fig7
cairo_pdf(file.path(supp.out, 'SupplementalFigure6.pdf'), family='Arial', width=13.2, height=8)
fig7
dev.off()
```

## Weighted co-occurence network analysis 

Using weighted co-occurrence network analysis, we identified 5 network modules of co-occurring ASVs. These networks were named after the top two most abundant taxa in the network, and the most central taxa in the network. The membership of all of the networks is in Additional file 10. To summarize these network modules, we summed the relative abundances of each taxa belonging to the network module compared the trajectories of these abundances over time between cases and controls. Change in the summed relative abundance of the 3 network modules not displayed in Figure 2B are shown in Supplemental Figure 7; the structure of the overall network is shown in Supplemental Figure 8. In Supplemental Figure 8, ASVs are nodes in the network. ASVs are colored by the module to which they were assigned and sized according to their abundance. The shape of the ASV node indicates if that ASV is more abundant in ECC cases (triangles) or controls (squares). Edges with a weight<0.1 have been eliminated from the network visualization for readability, otherwise edge width corresponds to the strength of the correlation between nodes. ASVs of special interest are labeled.  

Cases and controls differed in the summed relative abundance of these network modules and the distribution of community state types in both pre-incident (aditional file 9) and incident (Additional file 8) visits.

```{r unsupervised guilds networks - fig 2b}
#Pick top genera to color
topGenus<-c('Streptococcus', 'Neisseria', 'Veillonella', 'Fusobacterium', 'Haemophilus', 'Prevotella', 'Gemella', 'Actinomyces')
genus_colors=c(scales::hue_pal()(8), 'black')
names(genus_colors)=c(topGenus, 'Other')
#Pick top taxa to label 
topTax<- c('Fusobacterium periodonticum', 'Haemophilus parainfluenzae', 'Prevotella nanceinsis', 'Neisseria ASV9', 'Veillonella ASV5', 'Streptococcus ASV8', 'Prevotella histcola', 'Prevotella salivae', 'Prevotella melaninogenica', 'Streptococcus mutans', 'Fusobacterium nucleatum_subsp._animalis', imp_top%>%pull(ScientificName), hellinger_network$intraStats%>%group_by(module)%>%slice_max(n=2, order_by=kWithin)%>%pull(ScientificName))
#load extra fonts
extrafont::loadfonts()
#abundance info
my_df=ps_long%>% group_by(OTU,CaseEver)%>%summarise(m=mean(Abundance))%>%pivot_wider(names_from = CaseEver, values_from=m)%>%mutate(as=ifelse(Case>Control, 'Higher mean abundance case', 'Higher mean abundance controls'), m=ifelse(Case>Control, Case, Control))%>%mutate('name'=OTU)
#source tidygraph object creation from network object creation
nw<-hellinger_network
source(file.path(code.dir, 'Analysis', 'createTidyGraph.R'))


network_n=hellinger_network$module.data%>%filter(moduleColor %in% c('brown', 'yellow') & Visit<=IncidentVisit2)%>%group_by(moduleColor, Visit, Case)%>%summarise(n=n())%>%pivot_wider(names_from=Case, values_from=n)%>%mutate(modAbund=1, label=paste0('Case n=', Case, '\n Control n=', Control), Case='case')%>%filter(Visit %in% c(3, 5, 7, 8, 9, 10))%>%mutate(Visit=as.factor(Visit))

#plot change over time 
network_overtime_others=ggplot(hellinger_network$module.data%>%filter(Visit %in% c(3, 5, 7, 8, 9, 10) & !(moduleColor %in% c("brown", 'yellow')) & Visit<=IncidentVisit2), aes(x=as.factor(Visit), y=modAbund, color=as.factor(Case)))+stat_summary(geom='point', fun.y=mean)+facet_wrap(module_label3~., ncol=2, scales='free_y')+
    stat_summary(fun.data = mean_cl_boot, geom='errorbar', width=0.5)+#stat_compare_means(aes(label = ..p.signif..), size=5, hide.ns = TRUE, label.y=0.50)+
    geom_line(aes(group=UniquePersonID), alpha=0.1)+geom_smooth(aes(group=as.factor(Case), color=as.factor(Case)), se=F)+scale_color_manual(values=c("darkorange3", "royalblue3"), labels=c("ECC case", "Control"), name="")+theme_classic()+
    scale_x_discrete(name="Age (months)", labels=visit_labels2, expand=c(.1, .1))+ylab("Summed module relative abundance")+
    theme(text=element_text(size=14))+theme(axis.text.x=element_text(angle = 45, vjust = 0.5, hjust=.5))+
    #scale_y_continuous(breaks=c(0, .5, 1), minor_breaks = c(.25, .75))+
    theme(legend.position='bottom')+theme(strip.text = element_text(face='italic'))
```

```{r FIG 7, fig.cap='Summed network module relative abundance for the network modules in the COHRA2 cohort not shown in Figure 2 of main paper', fig.width=10, fig.height=8}
network_overtime_others
ggsave(filename=file.path(supp.out, 'SupplementalFigure7.eps'), network_overtime_others, units=c('in'), width=10, height=10, device=cairo_ps)
```

```{r FIG 8,  fig.cap='Structure of the entire network of ASVs, before separation into network modules', fig.width=8.5, fig.height=11}
entire_network<-gg_whole%>%ggraph('nicely')+geom_edge_link(aes(width=ifelse(weight>0.1, weight, NA)), alpha=1, color='grey')+geom_node_point(aes(color=factor(module_label3), size=m, shape=factor(as)))+geom_node_text(aes(label=ScientificNameDisplay), fontface='italic', family='Times', repel=TRUE, size=3)+guides(size=FALSE, edge_color=FALSE, color=guide_legend(nrow=3,byrow=TRUE), edge_width=FALSE)+scale_shape_manual('', values=c('triangle', 'square'))+scale_color_discrete(name='')+theme_graph(base_family='serif')+theme(legend.position = 'bottom')+scale_edge_width(range = c(1, 8))+scale_size(range=c(2, 8))+theme(strip.text.x = element_blank())+theme(text=element_text(family='serif', face='italic'))

entire_network

ggsave(filename=file.path(supp.out, 'SupplementalFigure8.eps'), entire_network, units=c('in'), width=10, height=10, device=cairo_ps)
```

```{r TABLE 7}
rm(entire_network)
module_membership<-left_join(hellinger_network$intraStats, hellinger_network$networkNames)%>%select(ASV, ScientificName, module_label2, module_label3,, module, kTotal, kWithin, kOut, kDiff)%>%group_by(module)%>%split(f=as.factor(.$module))
write.xlsx(module_membership, file=file.path(supp.out, 'AdditionalFile10.xlsx'), overwrite = T)
```




## Overlap between top ten separation taxa from community state typing and top two most abundant taxa from each network modules

Supplemental Figure 9 shows the overlap between the top two most abundant ASVs in each of the five network modules (ten ASVs) and the top ten most important ASVs for the separation of the community state types. There is significant overlap in these ASVs, with nine of the ten ASVs being the same. 

```{r unsupervised guilds vendiagram}
tax_contribution<-left_join(read.csv(file.path(output.out, 'CST', 'k6', 'k_6taxcontribution.csv'), col.names = c('ASV', 'Mean', paste0('CST', seq(1, 6)), 'diff', 'cdiff')), taxa)

#get the names of the top separation taxa (Cst)
#& top abundant network and hub taxa
top_seps=tax_contribution%>%head(n=10)%>%pull(ScientificName)
top_nets=c(hellinger_network$networkNames$TopTaxa1, hellinger_network$networkNames$TopTaxa2)
top_hubs=c(hellinger_network$networkNames$CentralTaxa1, hellinger_network$networkNames$CentralTaxa2)

#Get intersections for venndiagram 
middle=top_seps[top_seps %in% top_nets & top_seps %in% top_hubs]
seps_nets=top_seps[top_seps %in% top_nets]
seps_hubs=top_seps[top_seps %in% top_hubs]
nets_hubs=top_hubs[top_hubs %in% top_nets]
top_rf=imp_top$ScientificName[!str_detect(imp_top$ScientificName, 'mutans')]
#make venn diagram 
width_line<-1
#could do venn diagram comparing hubs to top abundant taxa +seps
#ggvenn(list('Top 10 CST \n separation taxa'=top_seps, 'Top 2 most abundant \n taxa from each WCN'=top_nets, 'Top 2 most central \n taxa from each WCN'=top_hubs))
#but for this paper, specifically, think the comparison between 
#top most abundant network hubs and top separation taxa is more important 
venn_comp=ggvenn(list('CST \n separation taxa'=top_seps, 'Network  \n abundant taxa'=top_nets, 'Network hub taxa'=top_hubs), show_elements = T, text_size=1.4, label_sep='\n')
```

```{r FIG 9, fig.cap='Venn diagram of top two ASVs in each of five network modules (10 ASVs) vs the top ten ASVs contributing to the separation of the community state types', fig.width=13, fig.height=8}
venn_comp

ggsave(filename=file.path(supp.out, 'SupplementalFigure9.eps'), venn_comp, units=c('in'), width=8, height=8)
```

## Robustness analysis

For the Dirichlet multinomial community state typing, we varied the choice of k community state types. When decreasing k to 5, the case-associated 12-month community state type was combined with one of the 2-month community state types (Supplemental Figure 10). When decreasing k to 4, the control-associated community state type was also collapsed. However, case- and control-associated community state types apparent at and after the 24-month visit were still apparent. The clustering of the samples through community state types was overall fairly robust, according to the rand index, the adjusted rand index and the adjusted mutual information index (Additional file 11).


```{r cst robust}
CST_compare<-left_join(left_join(left_join(CST_k6%>%mutate_at(vars(CST.8), function(x){paste0('k6_', x)})%>% rename_at(vars(CST_nocut:maxcomp.9), function(x){paste0("k6.", x)}),  CST_k5 %>%mutate_at(vars(CST.8), function(x){paste0('k5_', x)})%>% rename_at(vars(CST_nocut:maxcomp.9), function(x){paste0("k5.", x)})), CST_k4%>%mutate_at(vars(CST.8), function(x){paste0('k4_', x)}) %>% rename_at(vars(CST_nocut:maxcomp.9), function(x){paste0("k4.", x)})), MV %>% select(FoxCavID, UniquePersonID, UniqueID, BabySubjectID, Site, Case, IncidentVisit2, Visit))
compare=c('k4.CST.8', 'k5.CST.8', 'k6.CST.8')
comparedf=data.frame()
for(i in 1:(length(compare)-1)){
  for(j in 2:(length(compare))){
    if(i!=j){
    newrow=clustComp(CST_compare[, compare[i]], CST_compare[, compare[j]])%>%bind_rows()%>%
      mutate(comparison=paste0(compare[i], ' vs ', compare[j]), 
             cohort='COHRA2')
    comparedf=rbind(comparedf, newrow)
    }
  }
}

CST_compare_al<-CST_compare %>% group_by(Case, Visit, k6.CST.8, k5.CST.8, k4.CST.8) %>% dplyr::summarise(count=n())%>%mutate_at(vars(matches('CST.8')), ~ifelse(str_detect(., 'NA'), 'NA', .))

CST_compare_al<-to_lodes_form(CST_compare_al, axes=3:5, id='Cohort')

CST_compare_al$stratum=factor(CST_compare_al$stratum, levels=c('k6_3', 'k6_4', 'k6_6', 'k6_5', 'k6_1', 'k6_2', 'k5_4', 'k5_2', 'k5_5','k5_3', 'k5_1', 'k4_2', 'k4_1', 'k4_3', 'k4_4', 'NA'))

visit_labels=c('2 mo', '12 mo', '24 mo', '36 mo')
names(visit_labels)=c(3, 5, 7, 8)

cst_sens<-ggplot(CST_compare_al%>%filter(Visit %in% c(3, 5, 7, 8)),
       aes(x = x, stratum = stratum, alluvium = Cohort,
           y = count, fill=stratum)) +
    stat_stratum(width=.5)+geom_flow(alpha=.7)+geom_stratum(width=.5)+facet_grid(Case~Visit, labeller=labeller(Visit=visit_labels))+
    scale_fill_manual(name="", values=c("thistle", "lightcoral", "darkorange", "deepskyblue", "darkorange3","royalblue2", 'thistle', 'lightcoral', 'deepskyblue', 'darkorange3', 'royalblue2', 'thistle', 'lightcoral', 'darkorange3', 'royalblue2', 'grey'), labels=c(paste(rep('k=6: CST', 6), seq(1, 6)), paste(rep('k=5: CST', 5), seq(1, 5)), paste(rep('k=4: CST', 4), seq(1, 4)), 'Unassigned at P>0.8'))+scale_y_continuous('Number of samples assigned to CSTs')+scale_x_discrete(name='', labels=c('6 CSTs', '5 CSTs', '4 CSTs'))
```

```{r FIG 10, fig.cap='Sensitivity of the community state types in the COHRA2 cohort to the choice of k', fig.width=10}
cst_sens
ggsave(filename=file.path(supp.out, 'SupplementalFigure10.eps'), cst_sens, units=c('in'), width=8, height=8)
```

For the weighted co-occurrence network clusters, we varied the choice of normalization transform, comparing the Hellinger transform to the center-log ratio transform. When using the center-log ratio transform instead of the Hellinger transform, six network modules were identified instead of five. The network module with the most member taxa was split in two (Supplemental Figure 10), and some rearrangement of taxa membership existed between the other network modules, thus, clustering of the taxa across different choices of transform were better than random but not high. However, the ECC case status associated clusters (control enriched *Neisseria* ASV9-*H. parainfluenzae* network; case-enriched *Veillonella* ASV5-*Streptococcus* ASV8 network) were more stable and relatively robust (Supplemental Figure 11; Additional file 11).

```{r network robustness}
#a dataframe of the taxa in each network cluster across the two different transformations
compareModules=data.frame(taxa, 'clr'=clr_network$taxaColors, 'hellinger'=hellinger_network$taxaColors)
#rename the modules
hellingerColors=setNames(hellinger_network$networkNames$module_label3, hellinger_network$networkNames$module)
clrColors=setNames(clr_network$networkNames$module_label3, clr_network$networkNames$module)

compareModules$clr<-dplyr::recode(compareModules$clr, !!!clrColors)
compareModules$hellinger<-recode(compareModules$hellinger, !!!hellingerColors)
#convert to lode form for alluvial plot
compMod<-compareModules %>% group_by(hellinger, clr)%>%dplyr::summarise(freq=n())
compMod_lodes<-to_lodes_form(compMod, axes=1:2)

#reorder strata to make comparison easier
compMod_lodes$stratum<-factor(compMod_lodes$stratum, levels=c(
  "Haemophilus parainfluenzae & Neisseria ASV9 network\n (central taxa: Fusobacterium periodonticum)", 
  "Veillonella sp._HMT_780 & Granulicatella elegans network\n (central taxa: Alloprevotella sp._HMT_473)", 
  "Veillonella ASV5 & Streptococcus ASV8 network\n (central taxa: Lachnoanaerobaculum orale)", 
  "Prevotella histicola & Rothia ASV49 network\n (central taxa: Megasphaera micronuciformis)",
  "Gemella ASV2 & Leptotrichia shahii network\n (central taxa: Prevotella sp._HMT_317)",
  "Neisseria ASV12 & Veillonella ASV5 network\n (central taxa: Actinomyces ASV41)", 
  "Bergeyella sp._HMT_206 & Rothia ASV34 network\n (central taxa: Capnocytophaga ASV119)",
  "Streptococcus ASV1 & Neisseria ASV12 network\n (central taxa: Actinomyces ASV41)",
  "Streptococcus ASV1 & Gemella ASV2 network\n (central taxa: Prevotella ASV479)"
))

#plot taxa change
network_sensitivity<-ggplot(compMod_lodes, aes(x=x, stratum=stratum, fill=stratum, alluvium=alluvium, y=freq))+geom_flow()+geom_stratum()+scale_y_continuous(name='Amplicon sequence variants \n (273 total)')+scale_x_discrete(name='', labels=c('Hellinger \n network', 'Center log ratio \n network'))+scale_fill_discrete(name='Network modules', labels=paste0(c(rep('Hellinger & CLR: ', 2), 'Hellinger: ', 'CLR: ', 'Hellinger: ', 'CLR: ', 'CLR: ', 'Hellinger: ', 'CLR: '), levels(compMod_lodes$stratum)))+theme_classic()+theme(legend.text = element_text(face='italic'))

clr_greenyellowbrown<-ggplot(clr_network$module.data%>%filter(Visit %in% c(3, 5, 7, 8, 9, 10) & moduleColor %in% c("green", 'yellow', 'brown')), aes(x=as.factor(Visit), y=modAbund, color=as.factor(Case)))+stat_summary(geom='point', fun.y=mean)+facet_wrap(module_label3~., ncol=1, scales='free_y')+
    stat_summary(fun.data = mean_cl_boot, geom='errorbar', width=0.5)+#stat_compare_means(aes(label = ..p.signif..), size=5, hide.ns = TRUE, label.y=0.50)+
    geom_line(aes(group=UniquePersonID), alpha=0.1)+geom_smooth(aes(group=as.factor(Case), color=as.factor(Case)), se=F)+scale_color_manual(values=c("darkorange3", "royalblue3"), labels=c("ECC case", "Control"), name="")+theme_classic()+
    scale_x_discrete(name="Age (months)", labels=visit_labels2, expand=c(.1, .1))+ylab("Summed module relative abundance")+
    theme(text=element_text(size=12))+theme(axis.text.x=element_text(angle = 45, vjust = 0.5, hjust=.5))+
    #scale_y_continuous(breaks=c(0, .5, 1), minor_breaks = c(.25, .75))+
    theme(legend.position='bottom')+theme(strip.text = element_text(face='italic'))

hellinger_yellowbrowngreen<-ggplot(hellinger_network$module.data%>%filter(Visit %in% c(3, 5, 7, 8, 9, 10) & moduleColor %in% c("green", 'yellow', 'brown')), aes(x=as.factor(Visit), y=modAbund, color=as.factor(Case)))+stat_summary(geom='point', fun.y=mean)+facet_wrap(module_label3~., ncol=1, scales='free_y')+
    stat_summary(fun.data = mean_cl_boot, geom='errorbar', width=0.5)+#stat_compare_means(aes(label = ..p.signif..), size=5, hide.ns = TRUE, label.y=0.50)+
    geom_line(aes(group=UniquePersonID), alpha=0.1)+geom_smooth(aes(group=as.factor(Case), color=as.factor(Case)), se=F)+scale_color_manual(values=c("darkorange3", "royalblue3"), labels=c("ECC case", "Control"), name="")+theme_classic()+
    scale_x_discrete(name="Age (months)", labels=visit_labels2, expand=c(.1, .1))+ylab("Summed module relative abundance")+
    theme(text=element_text(size=12))+theme(axis.text.x=element_text(angle = 45, vjust = 0.5, hjust=.5))+
    #scale_y_continuous(breaks=c(0, .5, 1), minor_breaks = c(.25, .75))+
    theme(legend.position='bottom')+theme(strip.text = element_text(face='italic'))

network_sens2<-plot_grid(hellinger_yellowbrowngreen+stat_compare_means(label='p.signif', hide.ns=T)+ggtitle('Hellinger networks'), clr_greenyellowbrown+stat_compare_means(label='p.signif', hide.ns=T)+ggtitle('CLR networks'))
```

```{r FIG 11, fig.cap='Sensitivity of the network modules in the COHRA2 cohort to the choice of Hellinger vs center-log ratio transform', fig.width=16, fig.height=8}
network_sensall<-plot_grid(network_sensitivity, network_sens2, labels=c('A', 'B'))
cairo_pdf(file.path(supp.out, 'SupplementalFigure11.pdf'), family='Arial', width=20, height=8)
network_sensall
dev.off()
network_sensall
```

```{r ami ari ri}
#AMI, ARI, RI between 
clr_hellinger_cohra=clustComp(compareModules$clr, compareModules$hellinger)%>%bind_rows()%>%mutate(comparison='clr vs hellinger', cohort='COHRA2')

compareModules_trim=compareModules%>%filter(
  clr %in% c("Prevotella histicola & Rothia ASV49 network\n (central taxa: Megasphaera micronuciformis)", "Haemophilus parainfluenzae & Neisseria ASV9 network\n (central taxa: Fusobacterium periodonticum)", "Veillonella sp._HMT_780 & Granulicatella elegans network\n (central taxa: Alloprevotella sp._HMT_473)") & 
  hellinger %in% c("Veillonella ASV5 & Streptococcus ASV8 network\n (central taxa: Lachnoanaerobaculum orale)" , "Veillonella sp._HMT_780 & Granulicatella elegans network\n (central taxa: Alloprevotella sp._HMT_473)", "Haemophilus parainfluenzae & Neisseria ASV9 network\n (central taxa: Fusobacterium periodonticum)"))
clr_hellinger_cohra_trim=clustComp(compareModules_trim$clr, compareModules_trim$hellinger)%>%bind_rows()%>%mutate(comparison='clr vs hellinger: trimmed networks', cohort='COHRA2')

comparedf=rbind(comparedf, clr_hellinger_cohra, clr_hellinger_cohra_trim)

compare_cluster_table=comparedf%>%select(cohort, comparison, RI, ARI, AMI)
```

```{r TABLE 8}
write.xlsx(compare_cluster_table, file=file.path(supp.out, 'AdditionalFile11.xlsx'), overwrite = T)
```

## Replication analysis 

Supplemental figure 12 shows the full network modules from the Holgerson et al. cohort which were similar in composition to the ECC-associated networks in the COHRA2 cohort. Unlike in Main Figure 4, all of the edges are shown (in Main Figure 4C, edges with weights<0.03 are filtered out for visualization purposes). Notably, the protective *Haemophilus-Neisseria* network module had more nodes in the Holgerson et al. cohort than in the COHRA2 nested case-control cohort. The Holgerson et al. cohort had a much lower prevalence of ECC cases (10%) than the nested case-control cohort from COHRA2 (~50%) as a result of the design. Supplemental figure 13 shows other network modules from the Holgerson et al. cohort.  

```{r network and cst replicability}
prj_files<-c(list.files(file.path(data.out, 'PRJ', 'Network'), full.names = T), list.files(file.path(data.out, 'PRJ', 'CST'), full.names = T)[-4])
prj=new.env()
prj_files%>%purrr::map(function(file){load(file, envir = prj)})

load(file.path(data.out, 'PRJ', 'phyloseq', 'phyasv_postTax.Rdata'), envir = prj)
prj$ps_long=microbiome::transform(prj$prj.filter, 'compositional')%>%psmelt()%>%mutate(ScientificName=paste(Genus, ifelse(is.na(Species), OTU, Species), sep=' '))

nw<-prj$hellinger_network
my_df=prj$ps_long%>% group_by(OTU)%>%dplyr::summarise(m=mean(Abundance))%>%mutate(name=OTU)
source(file.path(code.dir, 'Analysis', 'createTidyGraph.R'))


br_yel_cohra<-hellinger_network$intraStats%>%filter(module%in% c('brown', 'yellow'))%>%pull(ScientificName)
by_cohra=br_yel_cohra[!str_detect(br_yel_cohra, 'ASV')]


matching_networks<-gg_whole%>%activate(nodes)%>%filter(module %in% c('turquoise', 'brown')) %>%ggraph('nicely')+geom_edge_link(aes(width=weight), alpha=0.8, color='grey')+geom_node_point(aes(color=GenusLabel, size=m))+geom_node_text(aes(label=ifelse(m>0.01, ScientificNameLabel, ifelse(ScientificName %in% by_cohra, ScientificNameLabel, ''))), fontface='italic', family='Times', repel=TRUE, size=3)+guides(size=FALSE, edge_color=FALSE, color=guide_legend(nrow=2,byrow=TRUE), edge_width=FALSE)+scale_color_manual('', values=genus_colors)+theme_graph(base_family='sans')+theme(legend.position = 'bottom')+scale_edge_width(range = c(0.1, 2))+theme(strip.text.x = element_blank())+facet_nodes(~module, scales='free')

other_networks<-gg_whole%>%activate(nodes)%>%filter(!(module %in% c('turquoise', 'brown'))) %>%ggraph('nicely')+geom_edge_link(aes(width=weight), alpha=0.8, color='grey')+geom_node_point(aes(color=GenusLabel, size=m))+geom_node_text(aes(label=ifelse(m>0.01, ScientificNameLabel, ifelse(ScientificName %in% by_cohra, ScientificNameLabel, ''))), fontface='italic', family='Times', repel=TRUE, size=3)+guides(size=FALSE, edge_color=FALSE, color=guide_legend(nrow=2,byrow=TRUE), edge_width=FALSE)+scale_color_manual('', values=genus_colors)+theme_graph(base_family='sans')+theme(legend.position = 'bottom')+scale_edge_width(range = c(0.1, 2))+theme(strip.text.x = element_blank())+facet_nodes(~module, scales='free')

cohra_names=hellinger_network$intraStats%>%select(module, ScientificName)%>%filter(!str_detect(ScientificName, 'ASV'))%>%group_by(module)%>%nest()

prj_names=prj$hellinger_network$intraStats%>%select(module, ScientificName)%>%filter(!str_detect(ScientificName, 'ASV'))%>%group_by(module)%>%nest()

shared_taxa=data.frame('prj'=character(), 'cohra'=character(), 'common'=character(), 'different'=character(), 
                       n_common=integer(), n_different=integer(), n_cohra=integer(), n_prj=integer())
for( i in 1:length(cohra_names$data)){
  for(j in 1: length(prj_names$data)){
    common=intersect(prj_names$data[[j]]$ScientificName, cohra_names$data[[i]]$ScientificName)
    n_common=length(common)
    different=setdiff(prj_names$data[[j]]$ScientificName, cohra_names$data[[i]]$ScientificName)
    n_diff=length(different)
    n_cohra=nrow(cohra_names$data[[i]]); n_prj=nrow(prj_names$data[[j]])
    new_row=data.frame(prj_names$module[[j]], cohra_names$module[[i]], paste(common, collapse='; '), 
              paste(different, collapse='; '), n_common, n_diff, n_cohra, n_prj)
    names(new_row)=names(shared_taxa)
    shared_taxa=rbind(shared_taxa, new_row)
  }
}
shared_taxa=shared_taxa%>%mutate(p_sharedC=n_common/n_cohra, p_sharedP=n_common/n_prj)

bVSb=shared_taxa %>%filter(cohra=='brown'& prj=='brown')
yVSt=shared_taxa %>%filter(cohra=='yellow'& prj=='turquoise')
```

```{r FIG 12, fig.cap='Network modules from the Holgerson et al. cohort that are similar in composition and strucutre to those observed in the COHRA2 cohort', fig.width=8, fig.height=11}
matching_networks
cairo_pdf(file.path(supp.out, 'SupplementalFigure12.pdf'), family='Arial', width=18, height=10)
matching_networks
dev.off()
```

```{r FIG 13, fig.cap='Other network modules identified in the Holgerson et al. cohort',fig.width=8, fig.height=11}
other_networks
cairo_pdf(file.path(supp.out, 'SupplementalFigure13.pdf'), family='Arial', width=18, height=10)
other_networks
dev.off()
```


```{r ami ari for between cohorts}
named_asvs=left_join(hellinger_network$intraStats%>%filter(!is.na(Species))%>%mutate(moduleCohra=paste0(module, 'cohra'))%>%select(ScientificName, moduleCohra), prj$hellinger_network$intraStats%>%filter(!is.na(Species))%>%mutate(modulePRJ=paste0(module, '_prj'))%>%select(ScientificName, modulePRJ))

named_asvs_nona=named_asvs%>%filter(!is.na(modulePRJ) & !is.na(moduleCohra))

named_asvs_trim=named_asvs%>%filter(modulePRJ %in%c('blue_prj', 'brown_prj', 'turquoise_prj') & moduleCohra %in% c('', ''))
```

 To compare the relatedness of the amplicon sequence variants assigned to various network modules across the COHRA and Holgerson cohort, we performed multiple sequence alignment of the amplicons using the R packages msa, using the ClustalW algorithim. We computed pairwise distances from the DNA sequences using the r function dist.dml from the r package phangorn, using the JC69 model. We created a neighbor joining tree using the phangorn function NJ, then fit a generalized time-reversible with gamma rate maximum likelihood tree using the neighbor joining tree as a starting point. We obtained 100 bootstrap values for the tree using bootstrap.pml and plotted the tree using ggtree and collapsed branches present in <50 of the bootstrapped trees. Supplemental Figure 14 shows bootstrapped phylogenetic trees for amplicons belonging to the A) *Neisseria* and B) *Veillonella* generas (respectively) from the Holgerson *et al.* and COHRA2 cohorts.

```{r network trees create data}
ps.f<-ps_meta_nodups
prj.f<-prj$prj.filter
#rename taxa
taxa_names(ps.f)<-paste0(taxa_names(ps.f), '_COHRA')
taxa_names(prj.f)<-paste0(taxa_names(prj.f), '_PRJ')
cohraasv<-taxa_names(ps.f)
prjasv<-taxa_names(prj.f)
#make new taxa frame
tax_table_merged<-as.matrix(rbind(left_join(as.data.frame(cbind(
  ps.f@tax_table, 'moduleColor'=hellinger_network$taxaColors, 'source'='COHRA')), hellinger_network$networkNames %>% select(moduleColor, module_label3))%>%
    mutate(modulesource=paste0(source,': ', module_label3)), left_join(as.data.frame(cbind(
      prj.f@tax_table, 'moduleColor'=prj$hellinger_network$taxaColors, 'source'="Holgerson et al.")), prj$hellinger_network$networkNames %>%
        select(moduleColor,module_label3))%>%
    mutate(modulesource=paste0(source,': ', module_label3))))

rownames(tax_table_merged)<-c(cohraasv, prjasv)
refseq_merged<-c(ps.f@refseq, prj.f@refseq)
otu_merged<-data.frame(matrix(rep(1, 701*4), nrow=4, ncol=701), row.names=paste0(rep('FAKE_', 4), 1:4))
sample_merged<-data.frame(matrix(rep(0, 4*4), nrow=4, ncol=4), row.names=paste0(rep('FAKE_', 4), 1:4))
names(otu_merged)<-c(taxa_names(ps.f), taxa_names(prj.f))
ps_relatedness<-phyloseq(otu_table(otu_merged, taxa_are_rows=F), tax_table(tax_table_merged), sample_data(sample_merged), refseq(refseq_merged))
```

```{r tree function and set seed, eval=F}
set.seed(03172021)
#this takes a long time to run, so ran once and saved plot as robject to laod back in. 
#make sure to run again if substantial changes up stream
#function based off of code here: 
#https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html#supervised_learning
#https://fuzzyatelin.github.io/bioanth-stats/module-24/module-24.html
#fn function from https://stackoverflow.com/questions/67705772/non-standard-evaluation-does-not-work-error-in-hsimpleerrormsg-call
fn <- function(ps, level, choice) {
        x <- paste0(level,"==","'",choice,"'")
        oldTax <- data.frame(tax_table(ps))
        newTax <- subset(oldTax, eval(parse(text=x)))
        tax_table(ps) <- tax_table(as.matrix(newTax))
        return(ps)
}

make_myTree=function(ps_object, level, choice, prob){
  g=fn(ps_object, level, choice)
  seqs=g@refseq
  #multiple sequence alignment
  mult=msa(seqs, method="ClustalW", type="dna", order="input")
  mult_dnabin<-as.DNAbin(mult)
  phang.align <- as.phyDat(mult, type="DNA", names=getSequence(seqs))
  dm <- dist.ml(phang.align) #distance ml for DNA
  dd<-dist.dna(mult_dnabin, model='N', pairwise.deletion = F)
  treeNJ_dd<-NJ(dd)
  treeNJ <- NJ(dm) # Note, tip order != sequence order
  fit = pml(treeNJ, data=phang.align)
  fitGTR <- update(fit, k=4, inv=0.2)
  fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE, rearrangement = "stochastic", control = pml.control(trace = 0))
  #bootstrap tree
  bs=bootstrap.pml(fitGTR, bs=100, optNni=TRUE, control=pml.control(trace=0), multicore = T)
  s<-plotBS(midpoint(fitGTR$tree), bs, p=prob, type='p')
  s2=ggtree::as.polytomy(s, feature='node.label', fun=function(x) as.numeric(x)<prob)
  #add tree to phyloseq object
  tree_collapsed<-merge_phyloseq(g, s2)
  return(tree_collapsed)
}


neisseria_tree=make_myTree(ps_relatedness, 'Genus',  'Neisseria', prob=50)

neis_tree<-ggtree(neisseria_tree)+geom_tiplab(aes(label=ifelse(label%in% c('ASV9_COHRA', 'ASV12_COHRA'), label, Species), color=modulesource))+geom_point(aes(color=modulesource), size=2)+scale_color_manual('Study source & network assignment', values=c("cornflowerblue", 'darkorange', "grey", "grey20", "blue4"))+theme(legend.position = c(0.7, 0.2))+theme_tree2()
d<-neis_tree$data
d<-d[!d$isTip,]
neis_tree_collapsed<-neis_tree+geom_text(data=d, aes(label=label), hjust=1.25)+theme(text=element_text(face='italic'))

veillonella_tree=make_myTree(ps_relatedness, 'Genus', 'Veillonella', prob=50)

veil_tree<-ggtree(veillonella_tree)+geom_tiplab(aes(label=ifelse(label %in% c('ASV4_COHRA', 'ASV5_COHRA'), label, Species), color=modulesource))+geom_point(aes(color=modulesource), size=2)+scale_color_manual('Study source & network assignment', values=c("grey", "darkgoldenrod", 'green', 'darkgreen', 'grey5', "grey20", 'darkorange'))+theme(legend.position = c(0.7, 0.2))+theme_tree2()
d<-veil_tree$data
d<-d[!d$isTip,]
veil_tree_collapsed<-veil_tree+geom_text(data=d, aes(label=label), hjust=1.25)+theme(text=element_text(face='italic'))

tree_figure=plot_grid(neis_tree_collapsed, veil_tree_collapsed, ncol=1, labels=c('A', 'B'))

save(tree_figure, file=file.path(og.data, 'treefigure.Rdata'))
```

```{r FIG 14, fig.cap='Phylogenetic trees showing relationships between A) Neisseria amplicons and B) Veillonella amplicons in the COHRA2 and Holgerson et al. cohorts', fig.width=12, fig.height=8}
load(file=file.path(og.data, 'treefigure.Rdata'))
tree_figure
```

```{r FIG 14 save}
ggsave(tree_figure, file=file.path(supp.out, 'SupplementalFigure14.pdf'), height=12, width=25)
```

## Multivariable sensitivity model 

Dietary sugar and oral hygiene are known risk factors for ECC. The COHRA2 cohort did collect a food frequency questionnaire, however, very few children had any reported consumption of high-sugar/cariogenic foods and beverages as early as 12-months of age (Additional File 12), when we could first observe an association between the salivary microbiome and early childhood caries. 



```{r case diet and toothbrushing}
mv_usc %>% 
  filter(Visit %in% c(3, 5, 7))%>%
  tbl_strata(strata=Visit, 
             ~.x %>% 
               tbl_summary(by=CaseEver, 
                           include=c(Q38i_Juice, 
                                     Q38l_Soda, 
                                     Q43j_EatDesserts, 
                                     Q43k_EatCandies, 
                                     WipeTeeth))%>%
               add_p())%>%
  as_flex_table()%>%
  save_as_docx(path=file.path(supp.out, 'AdditionalFile12.docx'))
```

```{additional diet investigation, eval=F}
cst_bydiet = cst_models %>% 
  filter(Visit %in% c(5, 7))%>%
  tbl_strata(strata=Visit, 
             ~.x %>% 
               droplevels()%>%
               tbl_summary(by=CST.8f, 
                           include=c(Q38i_Juice, 
                                     WipeTeeth))%>%
               add_p())

net_bydiet = network_models%>%
  filter(moduleColor %in% c('yellow', 'brown') & 
           Visit %in% c(3, 5, 7) &
           !is.na(Q38i_Juice))%>%
  ggplot(aes(x=Q38i_Juice, y=modAbund))+
  geom_violin()+geom_jitter()+
  stat_compare_means()+
  facet_grid(module_label2~Visit, 
             labeller=labeller(Visit=visit_labels2,
                               module_label2=label_wrap_gen(50)))+
  theme_bw()+
  theme(strip.text.y=element_text(face='italic'))+
  scale_x_discrete('Juice consumption frequency')+
  scale_y_continuous('Summed module abundance')
```


```{r salivary pH case status}

#move this to supplement, say n=10 missing data on salivary pH at visit 7, n=71 at visit 5
case_ph<-ggplot(MV %>% filter(Visit %in% c(5, 7) & Visit<=IncidentVisit2), aes(x=Case, y=SalivapH))+geom_boxplot()+stat_compare_means()+geom_jitter(alpha=0.5)+facet_grid(Visit~., labeller=labeller(Visit=visit_labels2))+theme_classic()+ylab("Salivary pH")+xlab("Early childhood caries case status")+theme(text=element_text(size=14))+theme(legend.position = c(0.1, 0.1))

#join together information on networks, CST, smutans, and diversity
netcst=left_join(left_join(hellinger_network$module.data, 
                               left_join(main_cst, cst_recodes)), MV)%>%
                  left_join(div%>%
                            dplyr::select(FoxCavID, Chao1, Shannon, 
                                        Abundance_Streptococcus_mutans,
                                        Prevalence_Streptococcus_mutans))%>%
  #create cst group variable for plotting
        dplyr::mutate(CSTGroup=case_when(CST.8%in% c(1, 2)~
                                           'Late childhood CSTs',
                                         CST.8 %in% c(3, 4)~
                                           "Infant CSTs", 
                                         CST.8 %in% c(5, 6)~
                                           'Early childhood CSTs'))
#filter to just visits of interest
netcst.57=netcst%>%
  filter(Visit %in% c(5, 7))

ggsave(case_ph, file=file.path(supp.out, 'SupplementalFigure15.pdf'), height=12, width=25)
```

```{r FIG 15, fig.cap='Salivary pH by case status among n=106 (12-months) and n=174 (24-months) pre-incident saliva samples', fig.width=12, fig.height=8}
case_ph
```

## Associations between unsupervised clusters from microbiome data and count of primary teeth/age at tooh emergence 

The *H. parainfluenzae & Neisseria ASV9* module did not correlate with the total number of primary teeth present at the 12- or 24-month visits, although the *Streptococcus ASV1 and Neisseria ASV12 module* did (Supplemental Figure 15). The *Veillonella ASV5* and *Streptococcus ASV8* network module was inversely correlated wtih count of primary teeth although this did not reach statistical significance. Interestingly, the *Streptococcus* ASVs that were a part of this network were closely related to *Streptococcus* species that have been previously reported to decrease after tooth emergence (see Dzidic  (2018). https://doi.org/10.1038/s41396-018-0204-z). *Streptococcus* ASV8 was likely *Streptococcus salivarius*, and *Streptococcus* ASV10 was closely related to *Streptococcus lactarius/peroris* (see Additional file 4 and supplemental figure 3), both of which Dzidic et al. reported decreased from 6 months to 24 months of age.   Similarly, although the mean number of primary teeth increased between community state types which were dominate at different times, the number of primary teeth did not significantly differ between the community state types that were codominant at any one time (Supplemental Figure 14). Among the subset of children recruited from Pennsylvania, data on the approximate age at first tooth emergence was available. In this subset, age at first tooth emergence was not associated with either WCN modules or CSTs.

```{r tooth number tooth emergence}
#cst to primary tooth count
cst_prim<-ggplot(netcst.57%>%filter(moduleColor=='brown' &!is.na(CSTlabels)), aes(x=CSTlabels, y=Prim_Tot_Teeth_Present, color=CSTlabels))+geom_boxplot()+geom_jitter()+facet_grid(Visit~., labeller=labeller(Visit=visit_labels2))+stat_compare_means(label.y=25)+theme_classic()+theme(axis.text.x = element_blank(), axis.title.x = element_blank())+theme(legend.position = 'bottom', legend.text = element_text(face='italic'))+scale_color_manual(name="Community state type", values=c("thistle","lightcoral", "deepskyblue",  "darkorange", "royalblue2", "darkorange3",  "royalblue4", "darkorange4", 'grey'), guide=guide_legend(nrow=2))+stat_compare_means(label.y=22, label='p.signif', comparisons=list(c("Gemella ASV2 - H. parainfluenzae - Neisseria ASV9", "Streptococcus ASV8 - Neisseria ASV12"), c("H. parainfluenzae - Neisseria ASV9", "Neisseria ASV12 - Veillonella ASV5"), c("Streptococcus ASV1 dominated w/ Gemella ASV2", "Streptoccous ASV1 dominated w/ G. elegans" )))+ggtitle('Primary teeth present & community state type')+ylab('Primary teeth present (count)')+theme(text=element_text(size=14))
#module -> teeth present
net_teeth<-ggplot(netcst.57, aes(x=Prim_Tot_Teeth_Present, y=modAbund))+geom_point()+facet_grid(Visit~module_label3, labeller = labeller(Visit=visit_labels2))+geom_smooth(color='red', method='lm')+theme_classic()+stat_cor()+ggtitle('Primary teeth present & network module abundance')+theme(strip.text.x = element_text(face='italic'))+xlab('# primary teeth')+ylab('Summed module relative abundance')+theme(text=element_text(size=14))
#case to teeth
case_teeth<-ggplot(MV%>%filter(Visit %in% c(5, 7)), aes(x=Case, y=Prim_Tot_Teeth_Present, color=Case))+geom_violin()+geom_boxplot(width=0.2)+geom_jitter()+facet_grid(Visit~., labeller=labeller(Visit=visit_labels2))+theme_classic()+stat_compare_means(label.y=25)+ggtitle('Primary teeth present & case status')+ylab('Primary teeth present (count)')+theme(text=element_text(size=14))+theme(legend.position='none')

prim_teeth<-plot_grid(plot_grid(case_teeth, cst_prim, nrow=1, labels=c('A', 'B'), rel_widths=c(0.3, 0.7), align='h', axis='tb'), net_teeth, nrow=2, labels=c('', 'C'))
```

```{r FIG 16, fig.cap='Relationships between unsupervised clusters and the number of primary teeth', fig.height=12, fig.width=25}
ggsave(prim_teeth, file=file.path(supp.out, 'SupplementalFigure16.pdf'), height=12, width=25)
prim_teeth
```

# Whole-genome shotgun metagenomics of incident samples reveals taxonomic and functional differences between incident case- and control-samples

## Volcano plot: taxa with abundances different between cases and controls in whole genome metagenome samples 

```{r volcano plot taxa}
#load data
load(file.path(data.out, 'MS', 'deseqresults.Rdata'))

cbPalette_volc=c('red', "#E69F00", "#56B4E9", "#009E73", "darkblue", "#0072B2", "purple", "#CC79A7", "#00FF00", 'darkred', "#000000")
taxa_of_interest=c('Neisseria perflava', 'Neisseria mucosa', 'Haemophilus parainfluenzae', 'Fusobacterium periodonticum', 'Fusobacterium nucleatum', 'Streptococcus sangunis', 'Prevotella salivae', 'Actinomyces graevenitzii', 'Streptococcus vestibularis', 'Prevotella histicola')
t3=t3%>%mutate(label=ifelse(Taxa %in% taxa_of_interest & padj<0.05, Taxa, ifelse((-log10(pvalue)>10)|(-log10(pvalue)>5 & SampleType=='saliva'), Taxa, '')))
taxa_volc=ggplot(t3%>%filter(Taxa!='Unmapped'),
       aes(x=log2FoldChange, y=-log10(pvalue),
           label=label,color=ColorCode))+
    geom_point()+geom_text_repel(max.overlaps = 500)+
    facet_wrap(~SampleType, scales='free_x')+
    scale_color_manual('', values=cbPalette_volc, na.translate=FALSE)+
    theme_classic()+xlab('\n \n log2 Fold change ')+
    theme(axis.line.x = element_line(arrow = 
                                         grid::arrow(length = unit(0.3, "cm"), ends = "both")))+
    theme(legend.position='bottom')

ggsave(taxa_volc, filename=file.path(supp.out, 'taxaVolcano.pdf'), units=c('in'), width=18, height=8)

t3_table=t3%>%select(-ColorCode)%>%arrange(padj)

t_nest=t3%>%filter(padj<0.05)%>%group_by(SampleType)%>%nest()
saliva_and_plaque=t_nest$data[[1]]$Taxa[t_nest$data[[1]]$Taxa %in% t_nest$data[[2]]$Taxa]
plaque_only=t_nest$data[[1]]$Taxa[!(t_nest$data[[1]]$Taxa %in% t_nest$data[[2]]$Taxa)]
saliva_only=t_nest$data[[2]]$Taxa[!(t_nest$data[[2]]$Taxa %in% t_nest$data[[1]]$Taxa)]
```

```{r TABLE 9}
write.xlsx(t3_table, file=file.path(supp.out, 'AdditionalFile13.xlsx'), overwrite=T)
```


```{r Table 10}
r3_table=r3%>%select(SampleType, KEGG, kname, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, PathAnnotation)%>%arrange(padj)
write.xlsx(r3_table, file=file.path(supp.out, 'AdditionalFile14.xlsx'), overwrite = T)
```

Additional file 13 contains the results from the DESEQ2 analysis on the taxa from WGS sequenced plaque and saliva samples, including the log2 fold change and adjusted *P* values. Supplemental figure 16 presents the same information as a volcano plot. Additional file 14 contains the results from the DESEQ2 analysis of KEGG orthologs from WGS sequenced plaque and saliva samples while additional file 15 contains the result from GSEA (gene set enrichment) analysis if the KEGG pathways. 

All the case enriched KEGG orthologs annotating to oxidative phosphorylation were found only in the genus Candida. For the 18 KEGG orthologs which were significantly different between cases and controls at a Benjamini-Hochberg corrected alpha of 0.05 and annotated to oxidative phosphorylation as their first KEGG pathway, additional file 16 shows the KEGG ortholog number, KEGG ortholog name, log2 fold change and all the taxonomic assignments for the contiguous sequences in which that KEGG ortholog was found. All the KEGG orthologs more abundant in cases were found only in either yeast or unclassified contiguous contigs, while the KEGG ortholgos more abundant in controls were found in bacterial species. Supplemental figure 17 shows the abundance of the species in which the KEGG orthologs were found.


```{r oxidative phosphorylation exploration}
load(file.path(data.out, 'MS', 'taxpathannotation.Rdata'))

op_case=bind_rows(op_taxa_list, .id='KEGG')%>%left_join(species_info$sample_table)%>%mutate(caseass=ifelse(KEGG %in% 'V/A-type', 'Control associated', 'Case associated'))%>%group_by(KEGG, SampleName)%>%filter(caseass=='Case associated')%>%ggplot(aes(x=Species, y=Abundance, color=CaseStatus))+geom_boxplot()+facet_wrap(~KEGG)+coord_flip()+theme_classic()+theme(axis.text.y = element_text(face='italic'))+scale_color_manual(values=c("darkorange3", "royalblue3"), labels=c("ECC case", "Control"), name="")

op_control=bind_rows(op_taxa_list, .id='KEGG')%>%left_join(species_info$sample_table)%>%mutate(caseass=ifelse(KEGG %in% 'V/A-type', 'Control associated', 'Case associated'))%>%group_by(KEGG, SampleName)%>%filter(caseass=='Control associated')%>%ggplot(aes(x=Species, y=Abundance, color=CaseStatus))+geom_boxplot()+facet_wrap(~KEGG)+coord_flip()+theme_classic()+theme(axis.text.y = element_text(face='italic'))+scale_color_manual(values=c("darkorange3", "royalblue3"), labels=c("ECC case", "Control"), name="")

op_plot=plot_grid(op_case, op_control, labels=c('A', 'B'), ncol=1)

op_table=left_join(r3%>%filter(padj<0.05 & PathAnnotation_01_3=='Oxidative phosphorylation')%>%mutate(p=word(kname, 1, 3))%>%mutate(p=word(p, 1))%>%select(KEGG, kname, p, log2FoldChange), lapply(op_taxa_list, function(x) x %>%select(Species)%>%unique())%>%bind_rows(.id='p')%>%group_by(p)%>%dplyr::summarise(Taxa=list(Species)))

```

```{r FIG 17, fig.cap='Abundance of taxa responsible for oxidative phosphorylation KEGG orthologs differentially abundant between cases and controls in plaque samples', fig.height=8, fig.width=10}
ggsave(op_plot, filename=file.path(supp.out, 'SupplementalFigure17.pdf'), units=c('in'), width=10, height=8)
op_plot
```


```{r enrichment analysis prep, eval=F}
pathways.list=keggList('pathway')
pathways.data=data.frame(pathway=names(pathways.list), pathwayNames=pathways.list, row.names=NULL)%>%mutate(pathway=gsub("path:", "", pathway))
pathway.codes <- sub("path:", "", names(pathways.list)) 
genes.by.pathway <- sapply(pathway.codes,
    function(pwid){
        pw <- tryCatch(keggGet(pwid), error=function(e) NULL)
        pw<-tryCatch(keggGet(pw[[1]]$KO_PATHWAY), error=function(e) NULL)
        if (is.null(pw[[1]]$ORTHOLOGY)) return(NA)
        pw2 <- names(pw[[1]]$ORTHOLOGY) # may need to modify this to c(FALSE, TRUE) for other organisms
        return(pw2)
    }
)

pathway.size=genes.by.pathway%>%map_dbl(~length(.x))

#get named log2FC in plaque and saliva samples
r3_split=r3%>%group_by(SampleType)%>%nest()%>%dplyr::mutate(my_genes=map(data, ~.x %>%pull(log2FoldChange)), my_gene_names=map(data, ~.x %>%pull(KEGG)))
r3_split=r3_split%>%mutate(my_genes=map2(my_gene_names, my_genes, ~.y%>%setNames(.x)))

#run fgsea for plaque and saliva samples separately 
r3_split = r3_split %>%
  mutate(my_fgsea=map(my_genes, 
                      ~fgsea(pathways=genes.by.pathway, 
                             stats=.x, 
                             minSize=1, 
                             maxSize=600)))

r3_split= r3_split %>% mutate(my_fgsea=
                      map(my_fgsea, 
                          ~left_join(.x, pathways.data)), 
                      my_sig=
                      map(my_fgsea, 
                          ~.x%>%
                          filter(padj<0.05)))

enrichment_table=r3_split%>%mutate(my_fgsea=map(my_fgsea, ~.x%>%arrange(padj))) %>% dplyr::select(SampleType, my_fgsea)%>%unnest(cols=c(my_fgsea))%>%bind_rows()
```

```{r gsea}
load(file.path(data.out, 'MS', 'keggpathways.Rdata'))


write.xlsx(enrichment_table, file.path(supp.out, 'AdditionalFile15.xlsx'), overwrite=T)

```

```{r Table 12}
write.xlsx(op_table, file=file.path(supp.out, 'AdditionalFile16.xlsx'), overwrite = T)
```
